// @pfizer/hlx-ui/src/components/viewers/StatusTag/StatusTag.stories.ts
import type { Meta, StoryObj } from '@storybook/vue3';
import readme from './README.md?raw';

import StatusTag from './StatusTag.vue';
import type { StatusVariant } from '@/enum/StatusVariant';

// More on how to set up stories at: https://storybook.js.org/docs/vue/writing-stories/introduction
const meta = {
  title: 'View/Status Tag',
  component: StatusTag,
  // This component will have an automatically generated docsPage entry: https://storybook.js.org/docs/vue/writing-docs/autodocs
  tags: ['autodocs'],
  parameters: {
    docs: {
      description: { component: readme },
    },
  },
  argTypes: {
    variant: {
      options: ['primary', 'accent', 'success', 'error', 'disabled'],
    },
  },
  args: {
    label: 'Success',
    variant: 'success' as StatusVariant,
  }, // default value
} satisfies Meta<typeof StatusTag>;

export default meta;
type Story = StoryObj<typeof meta>;
/*
 *👇 Render functions are a framework specific feature to allow you control on how the component renders.
 * See https://storybook.js.org/docs/vue/api/csf
 * to learn how to use render functions.
 */
export const Default: Story = {
  args: {},
};
