// @pfizer/hlx-ui/src/components/widgets/ContextMenu/ContextMenu.stories.ts
import type { Meta, StoryObj } from '@storybook/vue3';

import ContextMenu from './ContextMenu.vue';
import { provide } from 'vue';

// More on how to set up stories at: https://storybook.js.org/docs/vue/writing-stories/introduction
const meta = {
  title: 'Widget/Context Menu',
  component: ContextMenu,
  // This component will have an automatically generated docsPage entry: https://storybook.js.org/docs/vue/writing-docs/autodocs
  tags: ['autodocs'],
  argTypes: {},
  args: {
    items: [
      { title: 'First', key: '1' },
      { title: 'Second', key: '2', disabled: true },
      { title: 'Third', key: '3' },
      { title: 'link', key: '4', url: 'https://google.com' },
      { title: 'link disabled', key: '5', url: 'https://google.com', disabled: true },
    ],
  }, // default value
  render: (args) => ({
    components: { ContextMenu },
    setup() {
      provide('icon-font-config', {
        font: 'las',
        prefix: 'la-',
        iconMap: {},
      });
      return { args };
    },
    methods: {
      actionHandler(key: string) {
        alert(key);
      },
    },
    template: `
      <ContextMenu v-bind="args" @action="actionHandler">
        Click me!
      </ContextMenu>
      <ContextMenu v-bind="args" @action="actionHandler">
        Click me 2!
      </ContextMenu>
      <ContextMenu v-bind="args" mode="dropdown" @action="actionHandler">
        Dropdown
      </ContextMenu>
    `,
  }),
} satisfies Meta<typeof ContextMenu>;

export default meta;
type Story = StoryObj<typeof meta>;
/*
 *👇 Render functions are a framework specific feature to allow you control on how the component renders.
 * See https://storybook.js.org/docs/vue/api/csf
 * to learn how to use render functions.
 */
export const Default: Story = {
  args: {},
};
