// @pfizer/hlx-ui/src/components/viewers/Tooltip/Tooltip.stories.ts
import type { Meta, StoryObj } from '@storybook/vue3';
import readme from './README.md?raw';

import Tooltip from './Tooltip.vue';
import Flex from '../../layout/Flex';

// More on how to set up stories at: https://storybook.js.org/docs/vue/writing-stories/introduction
const meta = {
  title: 'View/Tooltip',
  component: Tooltip,
  // This component will have an automatically generated docsPage entry: https://storybook.js.org/docs/vue/writing-docs/autodocs
  tags: ['autodocs'],
  parameters: {
    docs: {
      description: { component: readme },
    },
  },
  argTypes: {},
  args: {
    placement: 'top',
    text: 'Lorem ipsum',
  }, // default value
  render: (args) => ({
    components: { Tooltip, Flex },
    setup() {
      return { args };
    },
    template: `
      <div style="height: 100px"></div>
      <Flex>
        <Tooltip text="Lorem ipsum">
          <button>Hover on me top</button>
        </Tooltip>
        <Tooltip text="Lorem ipsum" placement="right">
          <button>Hover on me right</button>
        </Tooltip>
        <Tooltip text="Lorem ipsum" placement="bottom">
          <button>Hover on me bottom</button>
        </Tooltip>
        <Tooltip text="Lorem ipsum" placement="left">
          <button>Hover on me left</button>
        </Tooltip>
      </Flex>
    `,
  }),
} satisfies Meta<typeof Tooltip>;

export default meta;
type Story = StoryObj<typeof meta>;
/*
 *👇 Render functions are a framework specific feature to allow you control on how the component renders.
 * See https://storybook.js.org/docs/vue/api/csf
 * to learn how to use render functions.
 */
export const Default: Story = {
  args: {},
};
