// @pfizer/hlx-ui/src/components/form/RadioField/RadioField.stories.ts
import type { Meta, StoryObj } from '@storybook/vue3';
import readme from './README.md?raw';

import RadioField from './RadioField.vue';

// More on how to set up stories at: https://storybook.js.org/docs/vue/writing-stories/introduction
const meta = {
  title: 'Form/Radio',
  component: RadioField,
  // This component will have an automatically generated docsPage entry: https://storybook.js.org/docs/vue/writing-docs/autodocs
  tags: ['autodocs'],
  parameters: {
    docs: {
      description: { component: readme },
    },
  },
  argTypes: {},
  args: {}, // default value
} satisfies Meta<typeof RadioField>;

export default meta;
type Story = StoryObj<typeof meta>;
/*
 *👇 Render functions are a framework specific feature to allow you control on how the component renders.
 * See https://storybook.js.org/docs/vue/api/csf
 * to learn how to use render functions.
 */
export const Default: Story = {
  args: {
    label: 'Input label',
    name: 'name',
    modelValue: '1',
    options: [
      { code: '1', label: 'First' },
      { code: '2', label: 'Second' },
      { code: '3', label: 'Third' },
    ],
  },
};
