// @pfizer/hlx-ui/src/components/layout/TopNavBar/TopNavBar.stories.ts
import type { Meta, StoryObj } from '@storybook/vue3';

import TopNavBar from './TopNavBar.vue';
import readme from './README.md?raw';

// More on how to set up stories at: https://storybook.js.org/docs/vue/writing-stories/introduction
const meta = {
  title: 'Layout/Top Nav Bar',
  component: TopNavBar,
  // This component will have an automatically generated docsPage entry: https://storybook.js.org/docs/vue/writing-docs/autodocs
  tags: ['autodocs'],
  parameters: {
    docs: {
      description: { component: readme },
    },
  },
  argTypes: {},
  args: {
    link: '#',
    logo: 'https://placehold.co/120x60',
  }, // default value
  render: (args) => ({
    components: { TopNavBar },
    setup() {
      return { args };
    },
    template: `
      <TopNavBar v-bind="args" >
        <a class="top-nav-bar__link active" href="#">Link 1</a>
        <a class="top-nav-bar__link" href="#">Link 1</a>
        <a class="top-nav-bar__link" href="#">Link 1</a>
        <a class="top-nav-bar__link" href="#">Link 1</a>
      </TopNavBar>
    `,
  }),
} satisfies Meta<typeof TopNavBar>;

export default meta;
type Story = StoryObj<typeof meta>;
/*
 *👇 Render functions are a framework specific feature to allow you control on how the component renders.
 * See https://storybook.js.org/docs/vue/api/csf
 * to learn how to use render functions.
 */
export const Default: Story = {
  args: {},
};
