// @pfizer/hlx-ui/src/components/viewers/Overlay/Overlay.stories.ts
import type { Meta, StoryObj } from '@storybook/vue3';
import readme from './README.md?raw';

import Overlay from './Overlay.vue';
import Flex from '../../layout/Flex';
import Preloader from '../Preloader';

// More on how to set up stories at: https://storybook.js.org/docs/vue/writing-stories/introduction
const meta = {
  title: 'View/Overlay',
  component: Overlay,
  // This component will have an automatically generated docsPage entry: https://storybook.js.org/docs/vue/writing-docs/autodocs
  tags: ['autodocs'],
  parameters: {
    docs: {
      description: { component: readme },
    },
  },
  argTypes: {},
  args: {
    isShow: false,
  }, // default value
  render: (args) => ({
    components: { Overlay, Flex, Preloader },
    setup() {
      return { args };
    },
    data() {
      return {
        show1: false,
        show2: false,
      };
    },
    methods: {
      btn1ClickHandler() {
        this.show1 = !this.show1;
      },
      btn2ClickHandler() {
        this.show2 = !this.show2;
      },
    },
    template: `
      <Flex>
        <Flex direction="column">
            <Overlay :is-show="show1">
              <template #default>
                <img src="https://placehold.co/600x400" alt="img">
              </template>
              <template #overlay>
                <Preloader />
              </template>
            </Overlay>
          <button type="button" @click="btn1ClickHandler">Show</button>
        </Flex>
        <Flex direction="column">
          <Overlay :is-show="show2">
            <template #default>
              <img src="https://placehold.co/600x400" alt="img">
            </template>
            <template #overlay>
              <div>
                <h4>Lorem Ipsum</h4>
                <p>Lorem ipsum dolor sit amet</p>
              </div>
            </template>
          </Overlay>
          <button type="button" @click="btn2ClickHandler">Show</button>
        </Flex>
      </Flex>
    `,
  }),
} satisfies Meta<typeof Overlay>;

export default meta;
type Story = StoryObj<typeof meta>;
/*
 *👇 Render functions are a framework specific feature to allow you control on how the component renders.
 * See https://storybook.js.org/docs/vue/api/csf
 * to learn how to use render functions.
 */
export const Default: Story = {
  args: {},
};
