// @pfizer/hlx-ui/src/components/viewers/ListGroup/ListGroup.stories.ts
import type { Meta, StoryObj } from '@storybook/vue3';
import readme from './README.md?raw';

import ListGroup from './ListGroup.vue';
import Flex from '../../layout/Flex';

// More on how to set up stories at: https://storybook.js.org/docs/vue/writing-stories/introduction
const meta = {
  title: 'View/ListGroup',
  component: ListGroup,
  // This component will have an automatically generated docsPage entry: https://storybook.js.org/docs/vue/writing-docs/autodocs
  tags: ['autodocs'],
  parameters: {
    docs: {
      description: { component: readme },
    },
  },
  argTypes: {},
  args: {
    items: ['First', 'Second', 'Third'],
    keygen: (item: unknown) => item as string,
  }, // default value
  render: (args) => ({
    components: { ListGroup, Flex },
    setup() {
      return { args };
    },
    template: `
      <Flex>
        <ListGroup v-bind="args" >
          <template #item="slotProps">
            {{ slotProps.item }}
          </template>
        </ListGroup>
        <ListGroup v-bind="args" variant="bordered" >
          <template #item="slotProps">
            {{ slotProps.item }}
          </template>
        </ListGroup>
      </Flex>
    `,
  }),
} satisfies Meta<typeof ListGroup>;

export default meta;
type Story = StoryObj<typeof meta>;
/*
 *👇 Render functions are a framework specific feature to allow you control on how the component renders.
 * See https://storybook.js.org/docs/vue/api/csf
 * to learn how to use render functions.
 */
export const Default: Story = {
  args: {},
};
