// @pfizer/hlx-ui/src/components/viewers/FormGroup/FormGroup.stories.ts
import type { Meta, StoryObj } from '@storybook/vue3';
import readme from './README.md?raw';

import FormGroup from './FormGroup.vue';

// More on how to set up stories at: https://storybook.js.org/docs/vue/writing-stories/introduction
const meta = {
  title: 'View/FormGroup',
  component: FormGroup,
  // This component will have an automatically generated docsPage entry: https://storybook.js.org/docs/vue/writing-docs/autodocs
  tags: ['autodocs'],
  parameters: {
    docs: {
      description: { component: readme },
    },
  },
  argTypes: {},
  args: {}, // default value
} satisfies Meta<typeof FormGroup>;

export default meta;
type Story = StoryObj<typeof meta>;
/*
 *👇 Render functions are a framework specific feature to allow you control on how the component renders.
 * See https://storybook.js.org/docs/vue/api/csf
 * to learn how to use render functions.
 */
export const Default: Story = {
  args: {},
  render: (args) => ({
    components: { FormGroup },
    setup() {
      return {
        args,
      };
    },
    template: `
      <FormGroup>
        <button>Click Me 1</button>
        <button>Click Me 2</button>
        <button>Click Me 3</button>
      </FormGroup>
    `,
  }),
};
