// @pfizer/hlx-ui/src/stories/WebFilesProto/LoginPage.stories.ts
import type { Meta, StoryObj } from '@storybook/vue3';

import TableView from '../../components/viewers/TableView/TableView.vue';
import Icon from '../../components/viewers/Icon/Icon.vue';

// More on how to set up stories at: https://storybook.js.org/docs/vue/writing-stories/introduction
const meta = {
  title: 'WebFilesProto/Login Page',
  component: TableView,
  // This component will have an automatically generated docsPage entry: https://storybook.js.org/docs/vue/writing-docs/autodocs
  tags: ['autodocs'],
  argTypes: {},
  args: {}, // default value
} satisfies Meta<typeof TableView>;

export default meta;
type Story = StoryObj<typeof meta>;
/*
 *👇 Render functions are a framework specific feature to allow you control on how the component renders.
 * See https://storybook.js.org/docs/vue/api/csf
 * to learn how to use render functions.
 */
export const Default: Story = {
  args: {
    header: [],
    data: [],
    keygen: () => 'test',
  },
  render: (args) => ({
    components: {
      TableView,
      Icon,
    },
    setup() {
      return {
        args,
      };
    },
    template: `
      <div class="page-auth">
        <img src="/logo.svg" alt="logo">
        <div class="page-auth__info-wrapper">
          <div class="page-auth__subtitle">Welcome to Web Files</div>
          <div class="page-auth__description">Web Files description</div>
        </div>
        <a
          href="#"
          class="button page-auth__button"
        >
          Login with Pfizer credentials
        </a>
        <div class="page-auth__bottom">
          <hr class="page-auth__bottom_underline">
          <img class="page-auth__bottom_logo" src="/pfizer-login.svg" alt="pfizer-login">
        </div>
      </div>
    `,
  }),
};
