// @pfizer/hlx-ui/src/stories/WebFilesProto/UserListPageDefault.stories.ts
import type { Meta, StoryObj } from '@storybook/vue3';

import TableView from '../../components/viewers/TableView/TableView.vue';
import Icon from '../../components/viewers/Icon/Icon.vue';
import ContextMenu from '../../components/widgets/ContextMenu/ContextMenu.vue';
import PageHeader from '../../components/layout/PageHeader/PageHeader.vue';
import Header from '@/stories/WebFilesProto/components/Header.vue';
import Panel from '../../components/layout/Panel/Panel.vue';
import Popup from '../../components/viewers/Popup/Popup.vue';
import SelectField from '@/components/form/SelectField';
import TextField from '@/components/form/TextField/TextField.vue';
import Checkbox from '@/components/form/Checkbox/Checkbox.vue';
import Grid from '@/components/layout/Grid';
import Paginator from '@/components/viewers/Paginator/Paginator.vue';
import { provide } from 'vue';
import { Sorting } from '../../enum/Sorting';

// More on how to set up stories at: https://storybook.js.org/docs/vue/writing-stories/introduction
const meta = {
  title: 'WebFilesProto/User List Page Default',
  component: TableView,
  // This component will have an automatically generated docsPage entry: https://storybook.js.org/docs/vue/writing-docs/autodocs
  tags: ['autodocs'],
  argTypes: {},
  args: {
    header: [
      { key: 'userName', label: 'User Name', sorting: Sorting.ASC },
      { key: 'roles', label: 'Roles', sorting: Sorting.DESC },
      { key: 'status', label: 'Status' },
      { key: 'sponsor', label: 'Sponsor' },
      { key: 'action', label: '' },
    ],
    data: [
      {
        userName: { email: 'qwerty@gmail.com', userId: '22223434' },
        roles: '',
        folders: ['Folder name 1', 'Folder name 1', 'Folder name 1', 'Folder name 1', 'Folder name 1', 'Folder name 1'],
        status: 'Published',
        sponsor: 'Sponsor name',
      },
      {
        userName: { email: 'qwerty@gmail.com', userId: '22223434' },
        roles: '',
        folders: ['Folder name 1', 'Folder name 1', 'Folder name 1', 'Folder name 1', 'Folder name 1', 'Folder name 1'],
        status: 'Published',
        sponsor: 'Sponsor name',
      },
      {
        userName: { email: 'qwerty@gmail.com', userId: '22223434' },
        roles: '',
        folders: ['Folder name 1', 'Folder name 1', 'Folder name 1', 'Folder name 1', 'Folder name 1', 'Folder name 1'],
        status: 'Published',
        sponsor: 'Sponsor name',
      },
      {
        userName: { email: 'qwerty@gmail.com', userId: '22223434' },
        roles: '',
        folders: ['Folder name 1', 'Folder name 1', 'Folder name 1', 'Folder name 1', 'Folder name 1', 'Folder name 1'],
        status: 'Published',
        sponsor: 'Sponsor name',
      },
      {
        userName: { email: 'qwerty@gmail.com', userId: '22223434' },
        roles: '',
        folders: ['Folder name 1', 'Folder name 1', 'Folder name 1', 'Folder name 1', 'Folder name 1', 'Folder name 1'],
        status: 'Published',
        sponsor: 'Sponsor name',
      },
    ],
    keygen: (param: unknown) => {
      const item = param as { name: string; age: string };

      return `${item.name}-${item.age}`;
    },
  }, // default value
} satisfies Meta<typeof TableView>;

export default meta;
type Story = StoryObj<typeof meta>;
/*
 *👇 Render functions are a framework specific feature to allow you control on how the component renders.
 * See https://storybook.js.org/docs/vue/api/csf
 * to learn how to use render functions.
 */
export const Default: Story = {
  args: {},
  render: (args) => ({
    components: {
      TableView,
      ContextMenu,
      Icon,
      PageHeader,
      SelectField,
      Grid,
      TextField,
      Header,
      Popup,
      Panel,
      Checkbox,
      Paginator,
    },
    setup() {
      provide('icon-font-config', {
        font: 'las',
        prefix: 'la-',
        iconMap: {},
      });

      return {
        args,
        items: [
          { title: 'View folder access', key: 'view', icon: 'eye' },
          { title: 'Edit', key: 'edit', icon: 'edit' },
          { title: 'Off-boarding user', key: 'delete', icon: 'trash' },
          { title: 'Activate user', key: 'activate', icon: 'unlock' },
        ],
        adminTools: [
          { code: 'userManagement', title: 'User management' },
          { code: 'auditLog', title: 'Audit log' },
          { code: 'backupLogs', title: 'Backup logs' },
          { code: 'folderRequests', title: 'Folder requests' },
        ],
        settings: [
          { code: 'userEdit', title: 'User edit' },
          { code: 'myAuditLog', title: 'My audit log' },
          { code: 'fileReschedule', title: 'File reschedule' },
          { code: 'fileSubscriptions', title: 'File subscriptions' },
          { code: 'search', title: 'Search' },
          { code: 'userInstruction', title: 'User instruction' },
        ],
        search: [
          { code: 'any', label: '-Any-' },
          { code: 'blocked', label: 'Blocked' },
          { code: 'active', label: 'Active' },
        ],
        temp: [
          { title: 'test', key: 'test' },
          { title: 'test', key: 'test' },
          { title: 'test', key: 'test' },
        ],
        showPopup: false,
      };
    },
    methods: {
      changeButtonState() {
        this.args.showPopup = !this.args.showPopup;
      },
    },
    template: `
      <div class="main">
        <Header></Header>
        <PageHeader title="User Management">
          <template #default>
            <button class="button" @click.prevent="changeButtonState">Add User</button>
          </template>
        </PageHeader>
        <Popup :model-value="args.showPopup" title="Add User">
          <template #content>
            <div class="inputs-block">
            <TextField name="name" label="Name"></TextField>
            <TextField name="email" label="Email"></TextField>
            <TextField name="sponsor" label="Sponsor"></TextField>
            </div>
            <div class="checkbox-panel">
            Roles
            <Checkbox label="Manager" name="Manager"></Checkbox>
            <Checkbox label="Editor" name="Editor"></Checkbox>
            <Checkbox label="Notify user about created account" name="notify"></Checkbox>
            </div>
          </template>
          <template #actions>
            <button class="secondary" @click.prevent="changeButtonState">
              No
            </button>
            <button @click.prevent="changeButtonState">Yes</button>
          </template>
        </Popup>
          <div class="container">
            <Panel class="panel">
            <TextField name="search" label="Search name"></TextField>
            <TextField name="search" label="Filter email" placeholder="Email"></TextField>
            <SelectField name="search" :options="search" label="User status"></SelectField>
            </Panel>
          </div>
        <div class="container">
          <div class="table-info"><p class="table-info__total">Total: 25</p><p class="link">Reset <Icon name="sync-alt"></Icon></p></div>
          <TableView v-bind="args" >
            <template #cell-userName="cell">
              <div>User ID {{cell.cell.userId}}</div>
              <div>User email {{cell.cell.email}}</div>
            </template>
            <template #cell-folders="cell">
              {{cell.cell.join(',')}}
            </template>
            <template #cell-action="cell">
              <ContextMenu :items="items" @action="changeButtonState">
                <Icon name="ellipsis-v" />
              </ContextMenu>
            </template>
          </TableView>
        </div>
        <Paginator class="container" total-pages="4" current-page="1" style="justify-content: center; margin-top: 20px"></Paginator>
      </div>
    `,
  }),
};
