// @pfizer/hlx-ui/src/components/viewers/Alert/Alert.stories.ts
import type { Meta, StoryObj } from '@storybook/vue3';
import readme from './README.md?raw';

import Alert from './Alert.vue';
import type { AlertType } from '@/enum/AlertType';

// More on how to set up stories at: https://storybook.js.org/docs/vue/writing-stories/introduction
const meta = {
  title: 'View/Alert',
  component: Alert,
  // This component will have an automatically generated docsPage entry: https://storybook.js.org/docs/vue/writing-docs/autodocs
  tags: ['autodocs'],
  parameters: {
    docs: {
      description: { component: readme },
    },
  },
  argTypes: {
    type: {
      control: 'select',
      options: ['success', 'error', 'info'],
    },
  },
  args: {
    id: 'alert',
    content: 'Lorem ipsum dolor sit amet',
    type: 'success' as AlertType,
    ttl: 99999,
  }, // default value
} satisfies Meta<typeof Alert>;

export default meta;
type Story = StoryObj<typeof meta>;
/*
 *👇 Render functions are a framework specific feature to allow you control on how the component renders.
 * See https://storybook.js.org/docs/vue/api/csf
 * to learn how to use render functions.
 */
export const Default: Story = {
  args: {},
};
