// @pfizer/hlx-ui/src/components/layout/Panel/Panel.stories.ts
import type { Meta, StoryObj } from '@storybook/vue3';

import Panel from './Panel.vue';

// More on how to set up stories at: https://storybook.js.org/docs/vue/writing-stories/introduction
const meta = {
  title: 'Layout/Panel',
  component: Panel,
  // This component will have an automatically generated docsPage entry: https://storybook.js.org/docs/vue/writing-docs/autodocs
  tags: ['autodocs'],
  argTypes: {},
  args: {},
  render: (args) => ({
    components: { Panel },
    setup() {
      return { args };
    },
    template: `
      <Panel v-bind="args" :title-large-gap="args.titleLargeGap" :single-column="args.singleColumn">
        <button>button</button> <button>button</button> <button>button</button> <button>button</button>
      </Panel>
    `,
  }),
} satisfies Meta<typeof Panel>;

export default meta;
type Story = StoryObj<typeof meta>;
/*
 *👇 Render functions are a framework specific feature to allow you control on how the component renders.
 * See https://storybook.js.org/docs/vue/api/csf
 * to learn how to use render functions.
 */
export const Default: Story = {
  args: {
    title: 'Page Title',
  },
  parameters: {
    slots: {
      default: {
        template: `<button>button</button> <button>button</button> <button>button</button> <button>button</button>`,
      },
    },
  },
};

export const SingleColumn: Story = {
  args: {
    title: 'Page Title',
    singleColumn: true,
  },
  parameters: {
    slots: {
      default: {
        template: `<button>button</button> <button>button</button> <button>button</button> <button>button</button>`,
      },
    },
  },
};
