// @pfizer/hlx-ui/src/components/layout/Grid/Grid.stories.ts
import type { Meta, StoryObj } from '@storybook/vue3';
import readme from './README.md?raw';

import Grid from './Grid.vue';
import Cell from './Cell.vue';

// More on how to set up stories at: https://storybook.js.org/docs/vue/writing-stories/introduction
const meta = {
  title: 'Layout/Grid',
  component: Grid,
  // This component will have an automatically generated docsPage entry: https://storybook.js.org/docs/vue/writing-docs/autodocs
  tags: ['autodocs'],
  parameters: {
    docs: {
      description: { component: readme },
    },
  },
  argTypes: {},
  args: {}, // default value
  render: (args) => ({
    components: { Grid, Cell },
    setup() {
      return { args };
    },
    template: `
      <Grid cols="4">
        <h1>test 1</h1>
        <Cell col="2/3" row="2/3">
          <h1>test 2</h1>
        </Cell>
        <h1>test 3</h1>
      </Grid>
    `,
  }),
} satisfies Meta<typeof Grid>;

export default meta;
type Story = StoryObj<typeof meta>;
/*
 *👇 Render functions are a framework specific feature to allow you control on how the component renders.
 * See https://storybook.js.org/docs/vue/api/csf
 * to learn how to use render functions.
 */
export const Default: Story = {
  args: {
    cols: 3,
  },
};
