// @pfizer/hlx-ui/src/components/layout/Flex/Flex.stories.ts
import type { Meta, StoryObj } from '@storybook/vue3';
import readme from './README.md?raw';

import Flex from './Flex.vue';

// More on how to set up stories at: https://storybook.js.org/docs/vue/writing-stories/introduction
const meta = {
  title: 'Layout/Flex',
  component: Flex,
  // This component will have an automatically generated docsPage entry: https://storybook.js.org/docs/vue/writing-docs/autodocs
  tags: ['autodocs'],
  parameters: {
    docs: {
      description: { component: readme },
    },
  },
  argTypes: {},
  args: {}, // default value
  render: (args) => ({
    components: { Flex },
    setup() {
      return { args };
    },
    template: `
      <Flex>
        <input type="text" />
        <input type="text" />
        <input type="text" />
      </Flex>
      <hr />
      <Flex direction="column" align-items="stretch">
        <input type="text" />
        <input type="text" />
        <input type="text" />
      </Flex>
      <hr />
      <Flex justify-content="center">
        <input type="text" />
        <input type="text" />
        <input type="text" />
      </Flex>
      <hr />
      <Flex direction="column" align-items="center">
        <input type="text" />
        <input type="text" />
        <input type="text" />
      </Flex>
    `,
  }),
} satisfies Meta<typeof Flex>;

export default meta;
type Story = StoryObj<typeof meta>;
/*
 *👇 Render functions are a framework specific feature to allow you control on how the component renders.
 * See https://storybook.js.org/docs/vue/api/csf
 * to learn how to use render functions.
 */
export const Default: Story = {
  args: {},
};
