// @pfizer/hlx-ui/src/components/layout/PageHeader/PageHeader.stories.ts
import type { Meta, StoryObj } from '@storybook/vue3';

import PageHeader from './PageHeader.vue';

// More on how to set up stories at: https://storybook.js.org/docs/vue/writing-stories/introduction
const meta = {
  title: 'Layout/Page Header',
  component: PageHeader,
  // This component will have an automatically generated docsPage entry: https://storybook.js.org/docs/vue/writing-docs/autodocs
  tags: ['autodocs'],
  argTypes: {},
  args: {},
  render: (args) => ({
    components: { PageHeader },
    setup() {
      return { args };
    },
    template: `
      <PageHeader v-bind="args" >
        <button>Create</button>
      </PageHeader>
    `,
  }),
} satisfies Meta<typeof PageHeader>;

export default meta;
type Story = StoryObj<typeof meta>;
/*
 *👇 Render functions are a framework specific feature to allow you control on how the component renders.
 * See https://storybook.js.org/docs/vue/api/csf
 * to learn how to use render functions.
 */
export const Default: Story = {
  args: {
    title: 'Page Title',
    backLink: '/',
  },
};

export const Simple: Story = {
  args: {
    title: 'Page Title',
    backLink: '',
  },
  render: (args) => ({
    components: { PageHeader },
    setup() {
      return { args };
    },
    template: `
      <PageHeader v-bind="args" />
    `,
  }),
};
