// @pfizer/hlx-ui/src/components/widgets/ServicesMenu/ServicesMenu.stories.ts
import type { Meta, StoryObj } from '@storybook/vue3';

import ServicesMenu from './ServicesMenu.vue';
import { provide } from 'vue';

// More on how to set up stories at: https://storybook.js.org/docs/vue/writing-stories/introduction
const meta = {
  title: 'Widget/Services Menu',
  component: ServicesMenu,
  // This component will have an automatically generated docsPage entry: https://storybook.js.org/docs/vue/writing-docs/autodocs
  tags: ['autodocs'],
  argTypes: {},
  args: {}, // default value
  render: (args) => ({
    components: { ServicesMenu },
    setup() {
      provide('icon-font-config', {
        font: 'las',
        prefix: 'la-',
        iconMap: {
          'square-menu-fill': 'angle-double-down',
          'square-menu': 'angle-down',
        },
      });
      return { args };
    },
    methods: {
      actionHandler(key: string) {
        alert(key);
      },
    },
    template: `
      <ServicesMenu />

    `,
  }),
} satisfies Meta<typeof ServicesMenu>;

export default meta;
type Story = StoryObj<typeof meta>;
/*
 *👇 Render functions are a framework specific feature to allow you control on how the component renders.
 * See https://storybook.js.org/docs/vue/api/csf
 * to learn how to use render functions.
 */
export const Default: Story = {
  args: {},
};
