// @pfizer/hlx-ui/src/components/viewers/Collapse/Collapse.stories.ts
import type { Meta, StoryObj } from '@storybook/vue3';
import readme from './README.md?raw';

import Collapse from './Collapse.vue';
import Flex from '../../layout/Flex';

// More on how to set up stories at: https://storybook.js.org/docs/vue/writing-stories/introduction
const meta = {
  title: 'View/Collapse',
  component: Collapse,
  // This component will have an automatically generated docsPage entry: https://storybook.js.org/docs/vue/writing-docs/autodocs
  tags: ['autodocs'],
  parameters: {
    docs: {
      description: { component: readme },
    },
  },
  argTypes: {},
  args: {}, // default value
  render: (args) => ({
    components: { Collapse, Flex },
    setup() {
      return { args };
    },
    template: `
      <Flex>
        <Collapse>
          Etiam tincidunt vehicula ligula in dapibus. Proin urna libero, auctor lobortis mi ultricies, tincidunt lacinia magna. Morbi non dui fringilla, aliquam neque et, pretium dolor. Vivamus pellentesque porttitor libero at dapibus. Donec tristique mauris ligula, non rutrum arcu posuere vitae. Phasellus nec mattis ex. Class aptent taciti sociosqu ad litora torquent per conubia nostra, per inceptos himenaeos. Aenean eget diam sed velit eleifend commodo. Etiam nec ornare orci, ut bibendum diam. Aliquam fermentum tempor sapien, nec elementum neque cursus ac. Praesent est enim, imperdiet eget felis ac, scelerisque malesuada augue. Aliquam tristique fermentum ante ac imperdiet. Vivamus consectetur sit amet nisi sit amet auctor. Vestibulum tortor justo, tincidunt eget commodo sed, condimentum at purus. Ut sed neque imperdiet, eleifend arcu non, viverra arcu.
        </Collapse>
      </Flex>
    `,
  }),
} satisfies Meta<typeof Collapse>;

export default meta;
type Story = StoryObj<typeof meta>;
/*
 *👇 Render functions are a framework specific feature to allow you control on how the component renders.
 * See https://storybook.js.org/docs/vue/api/csf
 * to learn how to use render functions.
 */
export const Default: Story = {
  args: {},
};
