// @pfizer/hlx-ui/src/components/viewers/Popup/Popup.stories.ts
import type { Meta, StoryObj } from '@storybook/vue3';
import readme from './README.md?raw';

import Popup from './Popup.vue';
import Modal from './Modal.vue';
import Confirmator from './Confirmator.vue';
import { provide } from 'vue';

// More on how to set up stories at: https://storybook.js.org/docs/vue/writing-stories/introduction
const meta = {
  title: 'View/Popup',
  component: Popup,
  // This component will have an automatically generated docsPage entry: https://storybook.js.org/docs/vue/writing-docs/autodocs
  tags: ['autodocs'],
  parameters: {
    docs: {
      description: { component: readme },
    },
  },
  argTypes: {},
  args: {}, // default value
} satisfies Meta<typeof Popup>;

export default meta;
type Story = StoryObj<typeof meta>;
/*
 *👇 Render functions are a framework specific feature to allow you control on how the component renders.
 * See https://storybook.js.org/docs/vue/api/csf
 * to learn how to use render functions.
 */
export const Default: Story = {
  args: { modelValue: false },
  render: (args) => ({
    components: { Popup, Modal, Confirmator },
    setup() {
      provide('icon-font-config', {
        font: 'las',
        prefix: 'la-',
        iconMap: {
          close: 'times',
        },
      });

      return {
        args,
      };
    },
    data() {
      return {
        isOpen: false,
        isOpenModal: false,
        isOpenLoadingModal: false,
        isConfirmatorOpened: false,
      };
    },
    methods: {
      btnClickHandler() {
        this.isOpen = true;
      },
      openModalHandler() {
        this.isOpenModal = true;
      },
      openLoadingModalHandler() {
        this.isOpenLoadingModal = true;
      },
      openConfirmatorHandler() {
        this.isConfirmatorOpened = true;
      },
    },
    template: `
      <button @click="btnClickHandler" type="button">Open popup</button>
      <button @click="openModalHandler" type="button">Open modal</button>
      <button @click="openLoadingModalHandler" type="button">Open loading modal</button>
      <button @click="openConfirmatorHandler" type="button">Open confirmator</button>
      <Popup v-model="isOpen" title="Popup title">
        <template #content>
          Lorem ipsum
        </template>
      </Popup>
      <Modal v-model="isOpenModal" title="Modal title" accept="Yes" cancel="No">
        Lorem ipsum
      </Modal>
      <Modal v-model="isOpenLoadingModal" title="Modal title loading" is-loading accept="Yes" cancel="No">
        Lorem ipsum
      </Modal>
      <Confirmator v-model="isConfirmatorOpened" title="Confirm changes" accept="Yes" cancel="No">
        Are you sure?
      </Confirmator>
    `,
  }),
};
