// @pfizer/hlx-ui/src/components/viewers/Skeleton/Skeleton.stories.ts
import type { Meta, StoryObj } from '@storybook/vue3';
import readme from './README.md?raw';

import Skeleton from './Skeleton.vue';
import Flex from '../../layout/Flex';

// More on how to set up stories at: https://storybook.js.org/docs/vue/writing-stories/introduction
const meta = {
  title: 'View/Skeleton',
  component: Skeleton,
  // This component will have an automatically generated docsPage entry: https://storybook.js.org/docs/vue/writing-docs/autodocs
  tags: ['autodocs'],
  parameters: {
    docs: {
      description: { component: readme },
    },
  },
  argTypes: {},
  args: {}, // default value
  render: (args) => ({
    components: { Skeleton, Flex },
    setup() {
      return { args };
    },
    template: `
      <Flex>
        <Skeleton />
        <Skeleton :w="5" variant="animated" />
        <Skeleton :w="7" variant="animated" />
        <Skeleton :w="3" variant="animated" />
      </Flex>
    `,
  }),
} satisfies Meta<typeof Skeleton>;

export default meta;
type Story = StoryObj<typeof meta>;
/*
 *👇 Render functions are a framework specific feature to allow you control on how the component renders.
 * See https://storybook.js.org/docs/vue/api/csf
 * to learn how to use render functions.
 */
export const Default: Story = {
  args: {},
};
