// @pfizer/hlx-ui/src/components/viewers/Menu/Menu.stories.ts
import type { Meta, StoryObj } from '@storybook/vue3';
import readme from './README.md?raw';

import Menu from './Menu.vue';
import Flex from '../../layout/Flex';
import useIcons from '@/providers/icons';

// More on how to set up stories at: https://storybook.js.org/docs/vue/writing-stories/introduction
const meta = {
  title: 'View/Menu',
  component: Menu,
  // This component will have an automatically generated docsPage entry: https://storybook.js.org/docs/vue/writing-docs/autodocs
  tags: ['autodocs'],
  parameters: {
    docs: {
      description: { component: readme },
    },
  },
  argTypes: {},
  args: {
    items: [
      {
        icon: 'edit',
        key: 'home',
        title: 'Home',
        url: '#',
      },
      {
        key: 'page1',
        title: 'Page1',
        url: '#',
      },
      {
        key: 'page2',
        title: 'Page 2',
        url: '#',
      },
    ],
  }, // default value
  render: (args) => ({
    components: { Menu, Flex },
    setup() {
      const icons = useIcons();
      icons.setupIconConfig({
        font: '',
        prefix: 'hi-',
        iconMap: {},
      });

      return { args };
    },
    template: `
      <Flex direction="column" align-items="flex-start">
        <Menu v-bind="args" />
        <hr>
        <Menu v-bind="args" variant="vertical" />
        <hr>
        <Menu v-bind="args" variant="breadcrumbs" />
        <hr>
        <Menu v-bind="args" variant="breadcrumbs">
          <template #divider>
            @_@
          </template>
        </Menu>
        <hr>
      </Flex>
    `,
  }),
} satisfies Meta<typeof Menu>;

export default meta;
type Story = StoryObj<typeof meta>;
/*
 *👇 Render functions are a framework specific feature to allow you control on how the component renders.
 * See https://storybook.js.org/docs/vue/api/csf
 * to learn how to use render functions.
 */
export const Default: Story = {
  args: {},
};
