// @pfizer/hlx-ui/dist/AutocompleteField.es.js
import { defineComponent as $, ref as C, watch as M, onUnmounted as A, onMounted as B, openBlock as y, createBlock as H, withCtx as D, createElementVNode as h, normalizeClass as w, withModifiers as N, withDirectives as O, unref as S, vModelText as U, createElementBlock as E, Fragment as j, renderList as q, toDisplayString as z, createCommentVNode as Q } from "vue";
import { _ as X } from "./FieldWrapper.vue_vue_type_script_setup_true_lang-69dbb8a3.mjs";
import { r as G, i as J } from "./isSymbol-b31e8206.mjs";
import { t as F } from "./toNumber-96596bda.mjs";
import "./Icon.vue_vue_type_script_setup_true_lang-bc90b628.mjs";
import "./Popup.vue_vue_type_style_index_0_lang-ca27e75b.mjs";
import "./dom-c1ddc50b.mjs";
var K = function() {
  return G.Date.now();
};
const x = K;
var P = "Expected a function", Y = Math.max, Z = Math.min;
function ee(g, r, u) {
  var l, s, p, a, n, o, m = 0, k = !1, d = !1, v = !0;
  if (typeof g != "function")
    throw new TypeError(P);
  r = F(r) || 0, J(u) && (k = !!u.leading, d = "maxWait" in u, p = d ? Y(F(u.maxWait) || 0, r) : p, v = "trailing" in u ? !!u.trailing : v);
  function e(t) {
    var c = l, _ = s;
    return l = s = void 0, m = t, a = g.apply(_, c), a;
  }
  function f(t) {
    return m = t, n = setTimeout(b, r), k ? e(t) : a;
  }
  function i(t) {
    var c = t - o, _ = t - m, L = r - c;
    return d ? Z(L, p - _) : L;
  }
  function T(t) {
    var c = t - o, _ = t - m;
    return o === void 0 || c >= r || c < 0 || d && _ >= p;
  }
  function b() {
    var t = x();
    if (T(t))
      return I(t);
    n = setTimeout(b, i(t));
  }
  function I(t) {
    return n = void 0, v && l ? e(t) : (l = s = void 0, a);
  }
  function R() {
    n !== void 0 && clearTimeout(n), m = 0, l = o = s = n = void 0;
  }
  function W() {
    return n === void 0 ? a : I(x());
  }
  function V() {
    var t = x(), c = T(t);
    if (l = arguments, s = this, o = t, c) {
      if (n === void 0)
        return f(o);
      if (d)
        return clearTimeout(n), n = setTimeout(b, r), e(o);
    }
    return n === void 0 && (n = setTimeout(b, r)), a;
  }
  return V.cancel = R, V.flush = W, V;
}
const ne = ["onClick"], te = ["placeholder", "disabled"], ae = {
  key: 0,
  class: "autocomplete__option-list"
}, le = { class: "autocomplete__option-wrapper" }, oe = ["onClick"], fe = /* @__PURE__ */ $({
  __name: "AutocompleteField",
  props: {
    label: {},
    error: {},
    disabled: { type: Boolean },
    className: {},
    name: {},
    options: {},
    modelValue: {},
    placeholder: {}
  },
  emits: ["update:modelValue", "query"],
  setup(g, { expose: r, emit: u }) {
    const l = g, s = C(null);
    r(s);
    const p = (e) => {
      s.value = e;
    }, a = C(!1), n = C(""), o = C(null), m = (e) => {
      a.value = !1, u("update:modelValue", e.code, l.name);
    }, k = () => {
      l.disabled || (a.value = !a.value);
    }, d = () => {
      a.value = !1;
    }, v = ee(() => {
      n.value.length >= 3 ? u("query", n.value) : n.value.length || u("update:modelValue", "", l.name);
    }, 500);
    return M(a, (e, f, i) => {
      o.value = i, e ? document.addEventListener("click", d) : document.removeEventListener("click", d);
    }), M(
      () => l.modelValue,
      (e) => {
        n.value = e;
      }
    ), A(() => {
      var e;
      return (e = o.value) == null ? void 0 : e.call(o);
    }), B(() => {
      l.modelValue && (n.value = l.modelValue);
    }), (e, f) => (y(), H(X, {
      label: e.label,
      error: e.error,
      "class-name": ` ${e.className || ""} ${a.value ? "active" : ""}`
    }, {
      default: D(() => [
        h("div", {
          ref: p,
          class: w([
            "autocomplete",
            {
              active: a.value,
              disabled: e.disabled
            }
          ])
        }, [
          h("div", {
            class: "autocomplete__header",
            onClick: N(k, ["stop"])
          }, [
            h("div", {
              class: w(["autocomplete__preview", { placeholder: !e.modelValue }])
            }, [
              O(h("input", {
                "onUpdate:modelValue": f[0] || (f[0] = (i) => n.value = i),
                type: "text",
                class: "autocomplete__input",
                placeholder: e.placeholder,
                disabled: e.disabled,
                onInput: f[1] || (f[1] = //@ts-ignore
                (...i) => S(v) && S(v)(...i))
              }, null, 40, te), [
                [U, n.value]
              ])
            ], 2)
          ], 8, ne),
          a.value ? (y(), E("div", ae, [
            h("div", le, [
              (y(!0), E(j, null, q(e.options, (i) => (y(), E("div", {
                key: i.code,
                class: "autocomplete__option",
                tabindex: "0",
                onClick: N((T) => m(i), ["stop", "prevent"])
              }, z(i.label), 9, oe))), 128))
            ])
          ])) : Q("", !0)
        ], 2)
      ]),
      _: 1
    }, 8, ["label", "error", "class-name"]));
  }
});
export {
  fe as default
};
