// @pfizer/hlx-ui/dist/ContextMenu.es.js
import { defineComponent as x, useCssVars as $, ref as d, openBlock as n, createElementBlock as l, Fragment as r, createElementVNode as N, normalizeClass as f, renderSlot as v, createVNode as i, createCommentVNode as u, withCtx as B, renderList as D, createBlock as k, createTextVNode as b, toDisplayString as h } from "vue";
import { _ as s } from "./Icon.vue_vue_type_script_setup_true_lang-bc90b628.mjs";
import { _ as M } from "./Flyout.vue_vue_type_style_index_0_lang-0b2ff5ea.mjs";
import { _ as T } from "./_plugin-vue_export-helper-dad06003.mjs";
const E = ["href"], H = ["disabled", "onClick"], S = /* @__PURE__ */ x({
  __name: "ContextMenu",
  props: {
    items: {},
    context: {},
    className: {},
    mode: { default: "default" },
    minWidth: { default: 0 }
  },
  emits: ["action"],
  setup(w, { emit: C }) {
    const c = w;
    $((t) => ({
      a7f067b2: t.minWidth
    }));
    const a = d(!1), m = d(void 0), p = d({
      top: 0,
      left: 0
    }), y = (t) => {
      t.preventDefault();
      const o = c.mode === "dropdown", e = t.currentTarget.getBoundingClientRect();
      p.value = {
        top: o ? e.top + e.height : e.top,
        left: e.left
      }, o && (m.value = `${e.width}`), a.value = !a.value;
    }, g = (t, o) => setTimeout(() => {
      C("action", t, c.context, o);
    }, 100);
    return (t, o) => (n(), l(r, null, [
      N("button", {
        class: f(["context-menu__btn", { show: a.value, invert: t.mode === "default", dropdown: t.mode === "dropdown" }]),
        onClick: y
      }, [
        v(t.$slots, "default", {}, () => [
          i(s, { name: "dots-vertical" })
        ], !0),
        t.mode === "dropdown" ? v(t.$slots, "dropdown-icon", { key: 0 }, () => [
          i(s, {
            name: "angle-down",
            "class-name": "dropdown-indicator"
          })
        ], !0) : u("", !0)
      ], 2),
      i(M, {
        modelValue: a.value,
        "onUpdate:modelValue": o[0] || (o[0] = (e) => a.value = e),
        position: p.value,
        width: m.value,
        "class-name": `context-menu ${t.className || ""} ${t.mode}`
      }, {
        default: B(() => [
          (n(!0), l(r, null, D(t.items, (e) => {
            var _;
            return n(), l(r, {
              key: e.key
            }, [
              (_ = e.url) != null && _.length ? (n(), l("a", {
                key: 0,
                href: e.disabled ? "#" : e.url,
                class: f(["button invert context-menu__item", { disabled: e.disabled }])
              }, [
                e.icon ? (n(), k(s, {
                  key: 0,
                  name: e.icon
                }, null, 8, ["name"])) : u("", !0),
                b(" " + h(e.title), 1)
              ], 10, E)) : (n(), l("button", {
                key: 1,
                class: "invert context-menu__item",
                disabled: e.disabled,
                onClick: (V) => g(e.key, V)
              }, [
                e.icon ? (n(), k(s, {
                  key: 0,
                  name: e.icon
                }, null, 8, ["name"])) : u("", !0),
                b(" " + h(e.title), 1)
              ], 8, H))
            ], 64);
          }), 128))
        ]),
        _: 1
      }, 8, ["modelValue", "position", "width", "class-name"])
    ], 64));
  }
});
const L = /* @__PURE__ */ T(S, [["__scopeId", "data-v-54e33b89"]]);
export {
  L as default
};
