// @pfizer/hlx-ui/dist/FieldWrapper.vue_vue_type_script_setup_true_lang-69dbb8a3.mjs
import { defineComponent as C, ref as r, openBlock as o, createBlock as s, resolveDynamicComponent as N, normalizeClass as $, withCtx as g, createElementBlock as n, createTextVNode as V, toDisplayString as t, createCommentVNode as i, createVNode as T, withModifiers as B, createElementVNode as E, renderSlot as H } from "vue";
import { _ as a } from "./Icon.vue_vue_type_script_setup_true_lang-bc90b628.mjs";
import { _ as M } from "./Popup.vue_vue_type_style_index_0_lang-ca27e75b.mjs";
import { g as S } from "./dom-c1ddc50b.mjs";
const q = {
  key: 0,
  ref: "setRef",
  class: "form-field__label"
}, D = {
  key: 0,
  class: "form-field__asterisk"
}, L = ["href", "download"], P = ["href"], R = ["innerHTML"], z = {
  key: 1,
  class: "form-field__error"
}, F = {
  key: 2,
  class: "form-field__description"
}, I = /* @__PURE__ */ C({
  __name: "FieldWrapper",
  props: {
    label: {},
    error: {},
    description: {},
    className: {},
    info: {},
    infoTitle: {},
    infoType: {},
    component: {},
    required: { type: Boolean }
  },
  setup(U, { expose: v }) {
    const w = r(null);
    v(w);
    const f = r(!1), d = r({
      top: 0,
      left: 0
    }), p = (e, l) => {
      l && (d.value = S(l.target)), f.value = e;
    };
    return (e, l) => {
      var m, h;
      return o(), s(N(e.component || "label"), {
        class: $(`form-field ${e.className || ""} ${(m = e.error) != null && m.length ? "invalid" : ""} ${(h = e.info) != null && h.length ? "has-tooltip" : ""}`)
      }, {
        default: g(() => {
          var u, c, _, k, y;
          return [
            e.label ? (o(), n("div", q, [
              V(t(e.label) + " ", 1),
              e.required ? (o(), n("span", D, " * ")) : i("", !0),
              (u = e.info) != null && u.length && e.infoType === "download" ? (o(), n("a", {
                key: 1,
                href: e.info,
                download: e.info
              }, [
                T(a, {
                  "class-name": "form-field__info",
                  name: "info"
                })
              ], 8, L)) : (c = e.info) != null && c.length && e.infoType === "link" ? (o(), n("a", {
                key: 2,
                href: e.info,
                target: "_blank"
              }, [
                T(a, {
                  "class-name": "form-field__info",
                  name: "info"
                })
              ], 8, P)) : (_ = e.info) != null && _.length ? (o(), s(a, {
                key: 3,
                "class-name": "form-field__info",
                name: "info",
                onClick: l[0] || (l[0] = B((b) => p(!0, b), ["stop", "prevent"]))
              })) : i("", !0),
              (k = e.info) != null && k.length || (y = e.infoTitle) != null && y.length ? (o(), s(M, {
                key: 4,
                "model-value": f.value,
                title: e.infoTitle,
                position: d.value,
                "hide-backdrop": "",
                "class-name": "tooltip",
                "onUpdate:modelValue": p
              }, {
                content: g(() => [
                  E("div", { innerHTML: e.info }, null, 8, R)
                ]),
                _: 1
              }, 8, ["model-value", "title", "position"])) : i("", !0)
            ], 512)) : i("", !0),
            H(e.$slots, "default"),
            e.error ? (o(), n("span", z, t(e.error), 1)) : i("", !0),
            e.description && !e.error ? (o(), n("span", F, t(e.description), 1)) : i("", !0)
          ];
        }),
        _: 3
      }, 8, ["class"]);
    };
  }
});
export {
  I as _
};
