// @pfizer/hlx-ui/dist/FileField.es.js
import { defineComponent as m, computed as c, openBlock as a, createBlock as f, withCtx as u, createElementVNode as n, createElementBlock as _, toDisplayString as h, createCommentVNode as b, createVNode as V } from "vue";
import { _ as g } from "./FieldWrapper.vue_vue_type_script_setup_true_lang-69dbb8a3.mjs";
import { _ as y } from "./Icon.vue_vue_type_script_setup_true_lang-bc90b628.mjs";
import "./Popup.vue_vue_type_style_index_0_lang-ca27e75b.mjs";
import "./dom-c1ddc50b.mjs";
const B = { class: "file-field__wrapper form-field__input" }, N = ["disabled", "placeholder", "multiple"], k = {
  key: 0,
  class: "file-field__placeholder"
}, j = /* @__PURE__ */ m({
  __name: "FileField",
  props: {
    label: {},
    error: {},
    disabled: { type: Boolean },
    className: {},
    name: {},
    modelValue: {},
    placeholder: {},
    info: {},
    description: {},
    multiple: { type: Boolean }
  },
  emits: ["update:modelValue"],
  setup(t, { emit: s }) {
    const l = t, d = (e) => {
      if (e) {
        if (e instanceof File)
          return e.name;
      } else
        return l.placeholder || "";
      return e instanceof FileList && (e = Array.from(e).map((i) => i.name)), e.map ? e.length ? e.join(", ") : l.placeholder : l.modelValue || l.placeholder;
    }, o = c(() => d(l.modelValue || l.placeholder)), p = (e) => {
      var r;
      const i = l.multiple ? e.target.files : (r = e.target.files) == null ? void 0 : r[0];
      s("update:modelValue", i, l.name);
    };
    return (e, i) => (a(), f(g, {
      label: e.label,
      error: e.error,
      info: e.info,
      description: e.description,
      "class-name": `field file-field ${e.className || ""}  ${e.disabled ? "disabled" : ""}`
    }, {
      default: u(() => [
        n("span", B, [
          n("input", {
            type: "file",
            disabled: e.disabled,
            class: "file-field__input form-field__input",
            placeholder: e.placeholder,
            multiple: e.multiple,
            onChange: p
          }, null, 40, N),
          o.value ? (a(), _("span", k, h(o.value), 1)) : b("", !0),
          V(y, { name: "file" })
        ])
      ]),
      _: 1
    }, 8, ["label", "error", "info", "description", "class-name"]));
  }
});
export {
  j as default
};
