// @pfizer/hlx-ui/dist/Flex-6cf567bd.mjs
import { defineComponent as n, useCssVars as s, openBlock as o, createElementBlock as a, normalizeClass as i, renderSlot as l } from "vue";
import { _ as r } from "./_plugin-vue_export-helper-dad06003.mjs";
const f = /* @__PURE__ */ n({
  __name: "Flex",
  props: {
    className: { default: "" },
    direction: { default: "row" },
    alignItems: { default: "center" },
    justifyContent: { default: "stretch" },
    mDirection: { default: (t) => t.direction },
    mAlignItems: { default: (t) => t.alignItems },
    mJustifyContent: { default: (t) => t.justifyContent }
  },
  setup(t) {
    return s((e) => ({
      "765817aa": e.mDirection,
      "6919b27e": e.mJustifyContent,
      "07e9f0f0": e.mAlignItems,
      "0c3bc332": e.direction,
      "0b6759e1": e.justifyContent,
      "14be7c5a": e.alignItems
    })), (e, m) => (o(), a("div", {
      class: i(["flex-layout", e.direction, e.className])
    }, [
      l(e.$slots, "default", {}, void 0, !0)
    ], 2));
  }
});
const d = /* @__PURE__ */ r(f, [["__scopeId", "data-v-a3f5c68f"]]);
export {
  d as F
};
