// @pfizer/hlx-ui/dist/Menu.es.js
import { provide as _, inject as g, defineComponent as f, openBlock as t, createBlock as s, resolveDynamicComponent as k, unref as o, normalizeProps as C, guardReactiveProps as y, withCtx as l, renderSlot as v, createElementBlock as u, Fragment as p, renderList as b, createVNode as N, normalizeClass as $, createCommentVNode as c, createTextVNode as m, toDisplayString as h } from "vue";
import { F as M } from "./Flex-6cf567bd.mjs";
import { _ as P } from "./Icon.vue_vue_type_script_setup_true_lang-bc90b628.mjs";
import "./_plugin-vue_export-helper-dad06003.mjs";
const d = "navigation";
function V() {
  const r = {
    link: {
      component: "a",
      propsMap: {
        url: "href"
      }
    }
  };
  return {
    setupNavConfig: (e) => {
      _(d, e);
    },
    navConfig: g(d) || r
  };
}
const B = /* @__PURE__ */ f({
  __name: "Link",
  props: {
    url: {}
  },
  setup(r) {
    const e = r, i = V(), n = {
      [i.navConfig.link.propsMap.url]: e.url
    };
    return (a, D) => (t(), s(k(o(i).navConfig.link.component), C(y({ ...a.$props, ...n })), {
      default: l(() => [
        v(a.$slots, "default")
      ]),
      _: 3
    }, 16));
  }
}), L = /* @__PURE__ */ f({
  __name: "Menu",
  props: {
    items: {},
    variant: { default: "default" }
  },
  setup(r) {
    return (e, i) => (t(), s(o(M), {
      "class-name": "menu",
      direction: e.variant === "vertical" ? "column" : void 0,
      "align-items": e.variant === "vertical" ? "flex-start" : "center",
      "m-align-items": e.variant === "vertical" ? "flex-start" : "center"
    }, {
      default: l(() => [
        (t(!0), u(p, null, b(e.items, (n, a) => (t(), u(p, {
          key: n.key
        }, [
          N(o(B), {
            url: n.disabled ? "#" : `${n.url}`,
            class: $(["button invert context-menu__item", { disabled: n.disabled }])
          }, {
            default: l(() => [
              n.icon ? (t(), s(o(P), {
                key: 0,
                name: n.icon
              }, null, 8, ["name"])) : c("", !0),
              m(" " + h(n.title), 1)
            ]),
            _: 2
          }, 1032, ["url", "class"]),
          a < e.items.length - 1 && e.variant === "breadcrumbs" ? v(e.$slots, "divider", { key: 0 }, () => [
            m("/")
          ]) : c("", !0)
        ], 64))), 128))
      ]),
      _: 3
    }, 8, ["direction", "align-items", "m-align-items"]));
  }
});
export {
  L as default
};
