// @pfizer/hlx-ui/dist/Modal.vue_vue_type_script_setup_true_lang-68462e3c.mjs
import { defineComponent as u, openBlock as a, createBlock as o, withCtx as l, renderSlot as f, createElementVNode as t, createTextVNode as c, toDisplayString as s, createCommentVNode as d, createElementBlock as k, Fragment as y } from "vue";
import { _ as V } from "./Popup.vue_vue_type_style_index_0_lang-ca27e75b.mjs";
import { _ as g } from "./Preloader.vue_vue_type_style_index_0_lang-f533812c.mjs";
import { _ as i } from "./Icon.vue_vue_type_script_setup_true_lang-bc90b628.mjs";
const _ = ["disabled"], b = ["disabled"], N = /* @__PURE__ */ u({
  __name: "Modal",
  props: {
    title: {},
    isLoading: { type: Boolean },
    accept: {},
    acceptIcon: {},
    cancel: {},
    cancelIcon: {},
    modelValue: { type: Boolean },
    className: {}
  },
  emits: ["update:modelValue", "accept"],
  setup(C, { emit: n }) {
    const m = (e) => n("update:modelValue", e), r = () => n("update:modelValue", !1), p = () => n("accept");
    return (e, I) => (a(), o(V, {
      "model-value": e.modelValue,
      title: e.title,
      "class-name": `${e.isLoading ? "loading" : ""} ${e.className}`,
      mode: "modal",
      "onUpdate:modelValue": m
    }, {
      content: l(() => [
        f(e.$slots, "default")
      ]),
      actions: l(() => [
        t("button", {
          type: "button",
          class: "secondary",
          disabled: e.isLoading,
          onClick: r
        }, [
          c(s(e.cancel || "Cancel") + " ", 1),
          e.cancelIcon ? (a(), o(i, {
            key: 0,
            name: e.cancelIcon
          }, null, 8, ["name"])) : d("", !0)
        ], 8, _),
        t("button", {
          type: "button",
          disabled: e.isLoading,
          onClick: p
        }, [
          e.isLoading ? (a(), o(g, {
            key: 0,
            "class-name": "invert"
          })) : (a(), k(y, { key: 1 }, [
            c(s(e.accept || "Accept"), 1)
          ], 64)),
          e.acceptIcon ? (a(), o(i, {
            key: 2,
            name: e.acceptIcon
          }, null, 8, ["name"])) : d("", !0)
        ], 8, b)
      ]),
      _: 3
    }, 8, ["model-value", "title", "class-name"]));
  }
});
export {
  N as _
};
