// @pfizer/hlx-ui/dist/PageHeader.es.js
import { defineComponent as c, openBlock as o, createElementBlock as n, normalizeClass as i, createVNode as l, createTextVNode as d, toDisplayString as r, createCommentVNode as _, createElementVNode as s, renderSlot as m } from "vue";
import { _ as p } from "./Icon.vue_vue_type_script_setup_true_lang-bc90b628.mjs";
const h = ["href"], k = { class: "page-header__title" }, f = { class: "page-header__buttons" }, L = /* @__PURE__ */ c({
  __name: "PageHeader",
  props: {
    title: {},
    backLink: {},
    className: {}
  },
  setup(b) {
    return (e, u) => {
      var a, t;
      return o(), n("div", {
        class: i(["page-header", e.className, "container", { "without-back": !((a = e.backLink) != null && a.length) }])
      }, [
        (t = e.backLink) != null && t.length ? (o(), n("a", {
          key: 0,
          class: "button invert page-header__back-btn",
          href: e.backLink
        }, [
          l(p, { name: "arrow-left" }),
          d(" " + r("Back"))
        ], 8, h)) : _("", !0),
        s("h3", k, r(e.title), 1),
        s("div", f, [
          m(e.$slots, "default")
        ])
      ], 2);
    };
  }
});
export {
  L as default
};
