// @pfizer/hlx-ui/dist/Paginator.es.js
import { defineComponent as y, computed as d, openBlock as s, createElementBlock as g, createElementVNode as f, withModifiers as u, createVNode as v, Fragment as m, createCommentVNode as c, renderList as _, normalizeClass as C, toDisplayString as b } from "vue";
import { _ as k } from "./Icon.vue_vue_type_script_setup_true_lang-bc90b628.mjs";
import { i as $, a as M, e as N } from "./_isIndex-3b6f7b82.mjs";
import { i as I } from "./isSymbol-b31e8206.mjs";
import { t as E } from "./toNumber-96596bda.mjs";
var R = Math.ceil, A = Math.max;
function D(t, n, e, a) {
  for (var p = -1, l = A(R((n - t) / (e || 1)), 0), r = Array(l); l--; )
    r[a ? l : ++p] = t, t += e;
  return r;
}
function F(t, n, e) {
  if (!I(e))
    return !1;
  var a = typeof n;
  return (a == "number" ? $(e) && M(n, e.length) : a == "string" && n in e) ? N(e[n], t) : !1;
}
var h = 1 / 0, V = 17976931348623157e292;
function P(t) {
  if (!t)
    return t === 0 ? t : 0;
  if (t = E(t), t === h || t === -h) {
    var n = t < 0 ? -1 : 1;
    return n * V;
  }
  return t === t ? t : 0;
}
function w(t) {
  return function(n, e, a) {
    return a && typeof a != "number" && F(n, e, a) && (e = a = void 0), n = P(n), e === void 0 ? (e = n, n = 0) : e = P(e), a = a === void 0 ? n < e ? 1 : -1 : P(a), D(n, e, a, t);
  };
}
var B = w();
const L = B, T = {
  key: 0,
  class: "paginator"
}, q = ["disabled"], z = {
  key: 0,
  class: "invert paginator__page",
  disabled: ""
}, G = ["onClick"], O = {
  key: 0,
  class: "invert paginator__page",
  disabled: ""
}, S = ["disabled"], U = /* @__PURE__ */ y({
  __name: "Paginator",
  props: {
    currentPage: {},
    totalPages: {}
  },
  emits: ["change"],
  setup(t, { emit: n }) {
    const e = t, a = d(() => e.currentPage < 2), p = d(() => e.currentPage >= e.totalPages), l = d(() => {
      const r = Math.max(e.currentPage - 2, 1), o = Math.min(e.currentPage + 2, e.totalPages + 1);
      return L(r, o);
    });
    return (r, o) => r.totalPages > 1 ? (s(), g("div", T, [
      f("button", {
        class: "invert paginator__prev-btn",
        disabled: a.value,
        onClick: o[0] || (o[0] = u((i) => n("change", r.currentPage < 2 ? 1 : r.currentPage - 1), ["prevent"]))
      }, [
        v(k, { name: "angle-left" })
      ], 8, q),
      r.currentPage > 3 ? (s(), g(m, { key: 0 }, [
        f("button", {
          class: "invert paginator__page",
          onClick: o[1] || (o[1] = u((i) => n("change", 1), ["prevent"]))
        }, "1"),
        r.currentPage > 4 ? (s(), g("button", z, "...")) : c("", !0)
      ], 64)) : c("", !0),
      (s(!0), g(m, null, _(l.value, (i) => (s(), g("button", {
        key: `page-${i}`,
        class: C(["invert", "paginator__page", { active: i === r.currentPage }]),
        onClick: u((X) => n("change", i), ["prevent"])
      }, b(i), 11, G))), 128)),
      r.currentPage < r.totalPages - 1 ? (s(), g(m, { key: 1 }, [
        r.currentPage < r.totalPages - 2 ? (s(), g("button", O, "...")) : c("", !0),
        f("button", {
          class: "invert paginator__page",
          onClick: o[2] || (o[2] = u((i) => n("change", r.totalPages), ["prevent"]))
        }, b(r.totalPages), 1)
      ], 64)) : c("", !0),
      f("button", {
        class: "invert paginator__next-btn",
        disabled: p.value,
        onClick: o[3] || (o[3] = u((i) => n("change", r.currentPage >= r.totalPages ? r.totalPages - 1 : r.currentPage + 1), ["prevent"]))
      }, [
        v(k, { name: "angle-right" })
      ], 8, S)
    ])) : c("", !0);
  }
});
export {
  U as default
};
