// @pfizer/hlx-ui/dist/RadioField.es.js
import { defineComponent as y, ref as d, computed as V, watch as g, openBlock as r, createBlock as _, withCtx as h, createElementVNode as t, createElementBlock as s, Fragment as B, renderList as N, withDirectives as k, vModelRadio as q, createTextVNode as w, toDisplayString as C } from "vue";
import { _ as R } from "./FieldWrapper.vue_vue_type_script_setup_true_lang-69dbb8a3.mjs";
import "./Icon.vue_vue_type_script_setup_true_lang-bc90b628.mjs";
import "./Popup.vue_vue_type_style_index_0_lang-ca27e75b.mjs";
import "./dom-c1ddc50b.mjs";
const T = ["disabled", "value", "name"], H = /* @__PURE__ */ y({
  __name: "RadioField",
  props: {
    label: {},
    error: {},
    disabled: { type: Boolean },
    className: {},
    name: {},
    info: {},
    infoType: {},
    options: {},
    modelValue: {},
    required: { type: Boolean }
  },
  emits: ["update:modelValue"],
  setup(m, { expose: u, emit: p }) {
    const a = m, l = d(a.modelValue), n = d(null);
    u(n);
    const f = (e) => {
      n.value = e;
    }, c = (e) => p("update:modelValue", e.target.value, a.name), v = V(() => a.name.replace(/[\][.]/gis, "_"));
    return g(a, () => {
      l.value = a.modelValue;
    }), (e, i) => (r(), _(R, {
      label: e.label,
      error: e.error,
      info: e.info,
      "info-type": e.infoType,
      required: e.required,
      "class-name": `radio-field ${e.className || ""} ${e.disabled ? "disabled" : ""} `,
      component: "div"
    }, {
      default: h(() => [
        t("div", {
          ref: f,
          class: "radio-field__list"
        }, [
          (r(!0), s(B, null, N(e.options, (o) => (r(), s("label", {
            key: o.code
          }, [
            k(t("input", {
              "onUpdate:modelValue": i[0] || (i[0] = (b) => l.value = b),
              type: "radio",
              class: "radio-field__input",
              disabled: e.disabled,
              value: o.code,
              name: v.value,
              onChange: c
            }, null, 40, T), [
              [q, l.value]
            ]),
            w(" " + C(o.label), 1)
          ]))), 128))
        ])
      ]),
      _: 1
    }, 8, ["label", "error", "info", "info-type", "required", "class-name"]));
  }
});
export {
  H as default
};
