// @pfizer/hlx-ui/dist/ServicesMenu.es.js
import { defineComponent as V, openBlock as d, createElementBlock as w, createElementVNode as f, toDisplayString as j, ref as h, onMounted as A, Fragment as M, normalizeClass as R, createBlock as _, createVNode as F, withCtx as N, renderList as W } from "vue";
import { _ as E } from "./Icon.vue_vue_type_script_setup_true_lang-bc90b628.mjs";
import { _ as H } from "./Flyout.vue_vue_type_style_index_0_lang-0b2ff5ea.mjs";
const v = (e, t) => t.some((n) => e instanceof n);
let C, L;
function $() {
  return C || (C = [
    IDBDatabase,
    IDBObjectStore,
    IDBIndex,
    IDBCursor,
    IDBTransaction
  ]);
}
function K() {
  return L || (L = [
    IDBCursor.prototype.advance,
    IDBCursor.prototype.continue,
    IDBCursor.prototype.continuePrimaryKey
  ]);
}
const I = /* @__PURE__ */ new WeakMap(), p = /* @__PURE__ */ new WeakMap(), m = /* @__PURE__ */ new WeakMap();
function z(e) {
  const t = new Promise((n, r) => {
    const o = () => {
      e.removeEventListener("success", i), e.removeEventListener("error", s);
    }, i = () => {
      n(l(e.result)), o();
    }, s = () => {
      r(e.error), o();
    };
    e.addEventListener("success", i), e.addEventListener("error", s);
  });
  return m.set(t, e), t;
}
function U(e) {
  if (I.has(e))
    return;
  const t = new Promise((n, r) => {
    const o = () => {
      e.removeEventListener("complete", i), e.removeEventListener("error", s), e.removeEventListener("abort", s);
    }, i = () => {
      n(), o();
    }, s = () => {
      r(e.error || new DOMException("AbortError", "AbortError")), o();
    };
    e.addEventListener("complete", i), e.addEventListener("error", s), e.addEventListener("abort", s);
  });
  I.set(e, t);
}
let D = {
  get(e, t, n) {
    if (e instanceof IDBTransaction) {
      if (t === "done")
        return I.get(e);
      if (t === "store")
        return n.objectStoreNames[1] ? void 0 : n.objectStore(n.objectStoreNames[0]);
    }
    return l(e[t]);
  },
  set(e, t, n) {
    return e[t] = n, !0;
  },
  has(e, t) {
    return e instanceof IDBTransaction && (t === "done" || t === "store") ? !0 : t in e;
  }
};
function k(e) {
  D = e(D);
}
function q(e) {
  return K().includes(e) ? function(...t) {
    return e.apply(g(this), t), l(this.request);
  } : function(...t) {
    return l(e.apply(g(this), t));
  };
}
function G(e) {
  return typeof e == "function" ? q(e) : (e instanceof IDBTransaction && U(e), v(e, $()) ? new Proxy(e, D) : e);
}
function l(e) {
  if (e instanceof IDBRequest)
    return z(e);
  if (p.has(e))
    return p.get(e);
  const t = G(e);
  return t !== e && (p.set(e, t), m.set(t, e)), t;
}
const g = (e) => m.get(e);
function J(e, t, { blocked: n, upgrade: r, blocking: o, terminated: i } = {}) {
  const s = indexedDB.open(e, t), a = l(s);
  return r && s.addEventListener("upgradeneeded", (c) => {
    r(l(s.result), c.oldVersion, c.newVersion, l(s.transaction), c);
  }), n && s.addEventListener("blocked", (c) => n(
    // Casting due to https://github.com/microsoft/TypeScript-DOM-lib-generator/pull/1405
    c.oldVersion,
    c.newVersion,
    c
  )), a.then((c) => {
    i && c.addEventListener("close", () => i()), o && c.addEventListener("versionchange", (u) => o(u.oldVersion, u.newVersion, u));
  }).catch(() => {
  }), a;
}
const Q = ["get", "getKey", "getAll", "getAllKeys", "count"], X = ["put", "add", "delete", "clear"], y = /* @__PURE__ */ new Map();
function P(e, t) {
  if (!(e instanceof IDBDatabase && !(t in e) && typeof t == "string"))
    return;
  if (y.get(t))
    return y.get(t);
  const n = t.replace(/FromIndex$/, ""), r = t !== n, o = X.includes(n);
  if (
    // Bail if the target doesn't exist on the target. Eg, getAll isn't in Edge.
    !(n in (r ? IDBIndex : IDBObjectStore).prototype) || !(o || Q.includes(n))
  )
    return;
  const i = async function(s, ...a) {
    const c = this.transaction(s, o ? "readwrite" : "readonly");
    let u = c.store;
    return r && (u = u.index(a.shift())), (await Promise.all([
      u[n](...a),
      o && c.done
    ]))[0];
  };
  return y.set(t, i), i;
}
k((e) => ({
  ...e,
  get: (t, n, r) => P(t, n) || e.get(t, n, r),
  has: (t, n) => !!P(t, n) || e.has(t, n)
}));
const Y = ["continue", "continuePrimaryKey", "advance"], S = {}, B = /* @__PURE__ */ new WeakMap(), O = /* @__PURE__ */ new WeakMap(), Z = {
  get(e, t) {
    if (!Y.includes(t))
      return e[t];
    let n = S[t];
    return n || (n = S[t] = function(...r) {
      B.set(this, O.get(this)[t](...r));
    }), n;
  }
};
async function* ee(...e) {
  let t = this;
  if (t instanceof IDBCursor || (t = await t.openCursor(...e)), !t)
    return;
  t = t;
  const n = new Proxy(t, Z);
  for (O.set(n, t), m.set(n, g(t)); t; )
    yield n, t = await (B.get(n) || t.continue()), B.delete(n);
}
function T(e, t) {
  return t === Symbol.asyncIterator && v(e, [IDBIndex, IDBObjectStore, IDBCursor]) || t === "iterate" && v(e, [IDBIndex, IDBObjectStore]);
}
k((e) => ({
  ...e,
  get(t, n, r) {
    return T(t, n) ? ee : e.get(t, n, r);
  },
  has(t, n) {
    return T(t, n) || e.has(t, n);
  }
}));
const te = 86400, b = "cache";
function x() {
  return Math.round(Date.now() / 1e3);
}
function ne() {
  return J("estimator-cache", 1, {
    upgrade(e) {
      e.createObjectStore(b);
    }
  });
}
async function re(e, t, n = te) {
  const r = await ne();
  let o = await r.get(b, e);
  const i = x();
  return (!o || o.ttl <= i) && (o = {
    value: await t(),
    ttl: x() + n
  }, await r.put(b, o, e)), o.value;
}
async function se() {
  return re("services-menu", async () => {
    const t = await (await fetch("https://webfiles.pfizer.com/file/3401ce30-3119-422a-a4f2-01fdcb60df4e")).json();
    return Promise.all(
      t.map(async (n) => {
        const r = await fetch(n.icon);
        return n.icon = await r.text(), n;
      })
    );
  });
}
const oe = ["href"], ie = ["innerHTML"], ce = { class: "services-menu__item-title" }, ae = /* @__PURE__ */ V({
  __name: "ServiceItem",
  props: {
    item: {}
  },
  setup(e) {
    return (t, n) => (d(), w("a", {
      href: t.item.url,
      target: "_blank",
      class: "services-menu__item"
    }, [
      f("span", {
        class: "services-menu__item-icon",
        innerHTML: t.item.icon
      }, null, 8, ie),
      f("span", ce, j(t.item.title), 1)
    ], 8, oe));
  }
}), ue = { class: "services-menu__inner" }, me = /* @__PURE__ */ V({
  __name: "ServicesMenu",
  setup(e) {
    const t = h([]), n = h(!1), r = h({
      top: 0,
      left: 0
    }), o = (i) => {
      i.stopPropagation(), i.preventDefault();
      const s = i.target.getBoundingClientRect();
      r.value = {
        top: s.top,
        left: s.left
      }, n.value = !n.value;
    };
    return A(async () => {
      t.value = await se();
    }), (i, s) => (d(), w(M, null, [
      f("button", {
        class: R(["invert", "services-menu__btn", { show: n.value }]),
        onClick: o
      }, [
        n.value ? (d(), _(E, {
          key: 0,
          name: "square-menu-fill",
          "class-name": "services-menu__icon"
        })) : (d(), _(E, {
          key: 1,
          name: "square-menu",
          "class-name": "services-menu__icon"
        }))
      ], 2),
      F(H, {
        modelValue: n.value,
        "onUpdate:modelValue": s[0] || (s[0] = (a) => n.value = a),
        position: r.value,
        "class-name": "services-menu"
      }, {
        default: N(() => [
          f("div", ue, [
            (d(!0), w(M, null, W(t.value, (a) => (d(), _(ae, {
              key: a.url,
              item: a
            }, null, 8, ["item"]))), 128))
          ])
        ]),
        _: 1
      }, 8, ["modelValue", "position"])
    ], 64));
  }
});
export {
  me as default
};
