// @pfizer/hlx-ui/dist/TableHead.vue_vue_type_script_setup_true_lang-57a51409.mjs
import { defineComponent as _, computed as C, openBlock as o, createElementBlock as p, normalizeClass as u, createElementVNode as k, unref as m, createBlock as n, createCommentVNode as a, createTextVNode as w, toDisplayString as S } from "vue";
import { _ as s } from "./Icon.vue_vue_type_script_setup_true_lang-bc90b628.mjs";
var e = /* @__PURE__ */ ((i) => (i.BOTH = "both", i.ASC = "asc", i.DESC = "desc", i))(e || {});
const f = {
  [e.BOTH]: e.ASC,
  [e.ASC]: e.DESC,
  [e.DESC]: e.ASC
}, b = { class: "table-view__heading-text" }, A = /* @__PURE__ */ _({
  __name: "TableHead",
  props: {
    item: {}
  },
  emits: ["sort"],
  setup(i, { emit: c }) {
    const t = i, l = C(() => {
      let r = "";
      return [e.ASC, e.DESC].indexOf(t.item.sorting) > -1 ? r = "sorted" : t.item.sorting === e.BOTH && (r = "can-sort"), `table-view__heading ${t.item.key} ${r}`;
    }), d = () => {
      t.item.sorting && c("sort", {
        key: t.item.key,
        direction: f[t.item.sorting]
      });
    };
    return (r, g) => (o(), p("th", {
      class: u(l.value),
      onClick: d
    }, [
      k("div", b, [
        t.item.sorting === m(e).ASC ? (o(), n(s, {
          key: 0,
          name: "arrow-up",
          "class-name": "table-view__sort-indicator"
        })) : a("", !0),
        t.item.sorting === m(e).DESC ? (o(), n(s, {
          key: 1,
          name: "arrow-down",
          "class-name": "table-view__sort-indicator"
        })) : a("", !0),
        t.item.sorting === m(e).BOTH ? (o(), n(s, {
          key: 2,
          name: "arrow-up-down",
          "class-name": "table-view__sort-indicator"
        })) : a("", !0),
        w(" " + S(r.item.label) + " ", 1),
        r.item.icon ? (o(), n(s, {
          key: 3,
          name: r.item.icon
        }, null, 8, ["name"])) : a("", !0)
      ])
    ], 2));
  }
});
export {
  A as _
};
