// @pfizer/hlx-ui/dist/TableView.es.js
import { defineComponent as u, openBlock as r, createElementBlock as s, createElementVNode as o, Fragment as a, renderList as l, createBlock as m, renderSlot as c, normalizeClass as k, createTextVNode as p, toDisplayString as w, createVNode as b } from "vue";
import { _ as h } from "./TableHead.vue_vue_type_script_setup_true_lang-57a51409.mjs";
import { _ as f } from "./Icon.vue_vue_type_script_setup_true_lang-bc90b628.mjs";
const v = { class: "table-view" }, $ = {
  key: 1,
  class: "table-view__row table-view__skel-tr"
}, g = {
  key: 0,
  class: "invert",
  disabled: ""
}, S = {
  key: 1,
  class: "table-view__skel"
}, A = /* @__PURE__ */ u({
  __name: "TableView",
  props: {
    header: {},
    data: {},
    emptyActionsIcon: {},
    keygen: { type: Function }
  },
  emits: ["sort"],
  setup(V, { emit: _ }) {
    const d = (e, i) => e[i], y = (e) => _("sort", e);
    return (e, i) => (r(), s("table", v, [
      o("thead", null, [
        o("tr", null, [
          (r(!0), s(a, null, l(e.header, (t) => (r(), m(h, {
            key: t.key,
            item: t,
            onSort: y
          }, null, 8, ["item"]))), 128))
        ])
      ]),
      o("tbody", null, [
        e.data.length ? (r(!0), s(a, { key: 0 }, l(e.data, (t) => (r(), s("tr", {
          key: e.keygen(t),
          class: "table-view__row"
        }, [
          c(e.$slots, "row", { row: t }, () => [
            (r(!0), s(a, null, l(e.header, (n) => (r(), s("td", {
              key: n.key,
              class: k(["table-view__cell", n.key])
            }, [
              c(e.$slots, `cell-${n.key}`, {
                cell: d(t, n.key),
                row: t
              }, () => [
                p(w(d(t, n.key)), 1)
              ])
            ], 2))), 128))
          ])
        ]))), 128)) : (r(), s("tr", $, [
          (r(!0), s(a, null, l(e.header, (t) => (r(), s("td", {
            key: `skel-${t.key}`,
            class: k(["table-view__skel-td", t.key])
          }, [
            t.key === "actions" ? (r(), s("button", g, [
              b(f, {
                name: e.emptyActionsIcon || "dots-horizontal"
              }, null, 8, ["name"])
            ])) : (r(), s("span", S))
          ], 2))), 128))
        ]))
      ])
    ]));
  }
});
export {
  A as default
};
