// @pfizer/hlx-ui/dist/Toast.vue_vue_type_style_index_0_lang-81b08ff2.mjs
import { defineComponent as r, ref as _, onMounted as m, onUnmounted as u, openBlock as p, createElementBlock as d, normalizeClass as f, createElementVNode as l, renderSlot as a, createVNode as v } from "vue";
import { _ as T } from "./Icon.vue_vue_type_script_setup_true_lang-bc90b628.mjs";
const y = { class: "toast__content" }, C = /* @__PURE__ */ r({
  __name: "Toast",
  props: {
    id: {},
    title: {},
    type: {},
    ttl: {}
  },
  emits: ["close"],
  setup(c, { emit: i }) {
    const s = c, e = _(!1);
    let t = 0;
    const n = () => {
      e.value = !1, setTimeout(() => {
        i("close", s);
      }, 400);
    };
    return m(() => {
      setTimeout(() => {
        e.value = !0;
      }, 10), t = window.setTimeout(() => {
        n();
      }, s.ttl);
    }), u(() => {
      t && clearTimeout(t);
    }), (o, h) => (p(), d("div", {
      class: f(["toast", o.type, { show: e.value }])
    }, [
      l("div", y, [
        a(o.$slots, "content")
      ]),
      l("button", {
        type: "button",
        class: "invert toast__btn-close",
        onClick: n
      }, [
        a(o.$slots, "close", {}, () => [
          v(T, { name: "close" })
        ])
      ])
    ], 2));
  }
});
export {
  C as _
};
