// @pfizer/hlx-ui/dist/useForm-e580410b.mjs
import { a as fe, r as A, i as M, S as L, c as j, b as J, f as _t } from "./isSymbol-b31e8206.mjs";
import { b as bt, e as ce, a as Je, c as $t, i as Xe } from "./_isIndex-3b6f7b82.mjs";
import { reactive as Tt, ref as k, computed as Ae, onUnmounted as At } from "vue";
var wt = Array.isArray;
const m = wt;
var Ot = /\.|\[(?:[^[\]]*|(["'])(?:(?!\1)[^\\]|\\.)*?\1)\]/, mt = /^\w*$/;
function Pt(e, t) {
  if (m(e))
    return !1;
  var r = typeof e;
  return r == "number" || r == "symbol" || r == "boolean" || e == null || fe(e) ? !0 : mt.test(e) || !Ot.test(e) || t != null && e in Object(t);
}
var St = A["__core-js_shared__"];
const ee = St;
var we = function() {
  var e = /[^.]+$/.exec(ee && ee.keys && ee.keys.IE_PROTO || "");
  return e ? "Symbol(src)_1." + e : "";
}();
function Et(e) {
  return !!we && we in e;
}
var jt = Function.prototype, xt = jt.toString;
function C(e) {
  if (e != null) {
    try {
      return xt.call(e);
    } catch {
    }
    try {
      return e + "";
    } catch {
    }
  }
  return "";
}
var Ct = /[\\^$.*+?()[\]{}|]/g, It = /^\[object .+?Constructor\]$/, Mt = Function.prototype, Lt = Object.prototype, Dt = Mt.toString, Ft = Lt.hasOwnProperty, Rt = RegExp(
  "^" + Dt.call(Ft).replace(Ct, "\\$&").replace(/hasOwnProperty|(function).*?(?=\\\()| for .+?(?=\\\])/g, "$1.*?") + "$"
);
function Ut(e) {
  if (!M(e) || Et(e))
    return !1;
  var t = bt(e) ? Rt : It;
  return t.test(C(e));
}
function Nt(e, t) {
  return e == null ? void 0 : e[t];
}
function I(e, t) {
  var r = Nt(e, t);
  return Ut(r) ? r : void 0;
}
var Bt = I(Object, "create");
const N = Bt;
function Gt() {
  this.__data__ = N ? N(null) : {}, this.size = 0;
}
function Ht(e) {
  var t = this.has(e) && delete this.__data__[e];
  return this.size -= t ? 1 : 0, t;
}
var Kt = "__lodash_hash_undefined__", zt = Object.prototype, Yt = zt.hasOwnProperty;
function qt(e) {
  var t = this.__data__;
  if (N) {
    var r = t[e];
    return r === Kt ? void 0 : r;
  }
  return Yt.call(t, e) ? t[e] : void 0;
}
var Wt = Object.prototype, Zt = Wt.hasOwnProperty;
function Jt(e) {
  var t = this.__data__;
  return N ? t[e] !== void 0 : Zt.call(t, e);
}
var Xt = "__lodash_hash_undefined__";
function Qt(e, t) {
  var r = this.__data__;
  return this.size += this.has(e) ? 0 : 1, r[e] = N && t === void 0 ? Xt : t, this;
}
function x(e) {
  var t = -1, r = e == null ? 0 : e.length;
  for (this.clear(); ++t < r; ) {
    var n = e[t];
    this.set(n[0], n[1]);
  }
}
x.prototype.clear = Gt;
x.prototype.delete = Ht;
x.prototype.get = qt;
x.prototype.has = Jt;
x.prototype.set = Qt;
function Vt() {
  this.__data__ = [], this.size = 0;
}
function X(e, t) {
  for (var r = e.length; r--; )
    if (ce(e[r][0], t))
      return r;
  return -1;
}
var kt = Array.prototype, er = kt.splice;
function tr(e) {
  var t = this.__data__, r = X(t, e);
  if (r < 0)
    return !1;
  var n = t.length - 1;
  return r == n ? t.pop() : er.call(t, r, 1), --this.size, !0;
}
function rr(e) {
  var t = this.__data__, r = X(t, e);
  return r < 0 ? void 0 : t[r][1];
}
function nr(e) {
  return X(this.__data__, e) > -1;
}
function ar(e, t) {
  var r = this.__data__, n = X(r, e);
  return n < 0 ? (++this.size, r.push([e, t])) : r[n][1] = t, this;
}
function w(e) {
  var t = -1, r = e == null ? 0 : e.length;
  for (this.clear(); ++t < r; ) {
    var n = e[t];
    this.set(n[0], n[1]);
  }
}
w.prototype.clear = Vt;
w.prototype.delete = tr;
w.prototype.get = rr;
w.prototype.has = nr;
w.prototype.set = ar;
var or = I(A, "Map");
const B = or;
function ir() {
  this.size = 0, this.__data__ = {
    hash: new x(),
    map: new (B || w)(),
    string: new x()
  };
}
function sr(e) {
  var t = typeof e;
  return t == "string" || t == "number" || t == "symbol" || t == "boolean" ? e !== "__proto__" : e === null;
}
function Q(e, t) {
  var r = e.__data__;
  return sr(t) ? r[typeof t == "string" ? "string" : "hash"] : r.map;
}
function ur(e) {
  var t = Q(this, e).delete(e);
  return this.size -= t ? 1 : 0, t;
}
function fr(e) {
  return Q(this, e).get(e);
}
function cr(e) {
  return Q(this, e).has(e);
}
function lr(e, t) {
  var r = Q(this, e), n = r.size;
  return r.set(e, t), this.size += r.size == n ? 0 : 1, this;
}
function O(e) {
  var t = -1, r = e == null ? 0 : e.length;
  for (this.clear(); ++t < r; ) {
    var n = e[t];
    this.set(n[0], n[1]);
  }
}
O.prototype.clear = ir;
O.prototype.delete = ur;
O.prototype.get = fr;
O.prototype.has = cr;
O.prototype.set = lr;
var gr = "Expected a function";
function le(e, t) {
  if (typeof e != "function" || t != null && typeof t != "function")
    throw new TypeError(gr);
  var r = function() {
    var n = arguments, o = t ? t.apply(this, n) : n[0], a = r.cache;
    if (a.has(o))
      return a.get(o);
    var i = e.apply(this, n);
    return r.cache = a.set(o, i) || a, i;
  };
  return r.cache = new (le.Cache || O)(), r;
}
le.Cache = O;
var dr = 500;
function pr(e) {
  var t = le(e, function(n) {
    return r.size === dr && r.clear(), n;
  }), r = t.cache;
  return t;
}
var hr = /[^.[\]]+|\[(?:(-?\d+(?:\.\d+)?)|(["'])((?:(?!\2)[^\\]|\\.)*?)\2)\]|(?=(?:\.|\[\])(?:\.|\[\]|$))/g, vr = /\\(\\)?/g, yr = pr(function(e) {
  var t = [];
  return e.charCodeAt(0) === 46 && t.push(""), e.replace(hr, function(r, n, o, a) {
    t.push(o ? a.replace(vr, "$1") : n || r);
  }), t;
});
const _r = yr;
function br(e, t) {
  for (var r = -1, n = e == null ? 0 : e.length, o = Array(n); ++r < n; )
    o[r] = t(e[r], r, e);
  return o;
}
var $r = 1 / 0, Oe = L ? L.prototype : void 0, me = Oe ? Oe.toString : void 0;
function Qe(e) {
  if (typeof e == "string")
    return e;
  if (m(e))
    return br(e, Qe) + "";
  if (fe(e))
    return me ? me.call(e) : "";
  var t = e + "";
  return t == "0" && 1 / e == -$r ? "-0" : t;
}
function Tr(e) {
  return e == null ? "" : Qe(e);
}
function Ve(e, t) {
  return m(e) ? e : Pt(e, t) ? [e] : _r(Tr(e));
}
var Ar = 1 / 0;
function ke(e) {
  if (typeof e == "string" || fe(e))
    return e;
  var t = e + "";
  return t == "0" && 1 / e == -Ar ? "-0" : t;
}
function wr(e, t) {
  t = Ve(t, e);
  for (var r = 0, n = t.length; e != null && r < n; )
    e = e[ke(t[r++])];
  return r && r == n ? e : void 0;
}
function K(e, t, r) {
  var n = e == null ? void 0 : wr(e, t);
  return n === void 0 ? r : n;
}
var Or = function() {
  try {
    var e = I(Object, "defineProperty");
    return e({}, "", {}), e;
  } catch {
  }
}();
const Pe = Or;
function et(e, t, r) {
  t == "__proto__" && Pe ? Pe(e, t, {
    configurable: !0,
    enumerable: !0,
    value: r,
    writable: !0
  }) : e[t] = r;
}
var mr = Object.prototype, Pr = mr.hasOwnProperty;
function ge(e, t, r) {
  var n = e[t];
  (!(Pr.call(e, t) && ce(n, r)) || r === void 0 && !(t in e)) && et(e, t, r);
}
function Sr(e, t, r, n) {
  if (!M(e))
    return e;
  t = Ve(t, e);
  for (var o = -1, a = t.length, i = a - 1, u = e; u != null && ++o < a; ) {
    var f = ke(t[o]), c = r;
    if (f === "__proto__" || f === "constructor" || f === "prototype")
      return e;
    if (o != i) {
      var h = u[f];
      c = n ? n(h, f, u) : void 0, c === void 0 && (c = M(h) ? h : Je(t[o + 1]) ? [] : {});
    }
    ge(u, f, c), u = u[f];
  }
  return e;
}
function R(e, t, r) {
  return e == null ? e : Sr(e, t, r);
}
function Er() {
  this.__data__ = new w(), this.size = 0;
}
function jr(e) {
  var t = this.__data__, r = t.delete(e);
  return this.size = t.size, r;
}
function xr(e) {
  return this.__data__.get(e);
}
function Cr(e) {
  return this.__data__.has(e);
}
var Ir = 200;
function Mr(e, t) {
  var r = this.__data__;
  if (r instanceof w) {
    var n = r.__data__;
    if (!B || n.length < Ir - 1)
      return n.push([e, t]), this.size = ++r.size, this;
    r = this.__data__ = new O(n);
  }
  return r.set(e, t), this.size = r.size, this;
}
function T(e) {
  var t = this.__data__ = new w(e);
  this.size = t.size;
}
T.prototype.clear = Er;
T.prototype.delete = jr;
T.prototype.get = xr;
T.prototype.has = Cr;
T.prototype.set = Mr;
var Lr = "__lodash_hash_undefined__";
function Dr(e) {
  return this.__data__.set(e, Lr), this;
}
function Fr(e) {
  return this.__data__.has(e);
}
function q(e) {
  var t = -1, r = e == null ? 0 : e.length;
  for (this.__data__ = new O(); ++t < r; )
    this.add(e[t]);
}
q.prototype.add = q.prototype.push = Dr;
q.prototype.has = Fr;
function Rr(e, t) {
  for (var r = -1, n = e == null ? 0 : e.length; ++r < n; )
    if (t(e[r], r, e))
      return !0;
  return !1;
}
function Ur(e, t) {
  return e.has(t);
}
var Nr = 1, Br = 2;
function tt(e, t, r, n, o, a) {
  var i = r & Nr, u = e.length, f = t.length;
  if (u != f && !(i && f > u))
    return !1;
  var c = a.get(e), h = a.get(t);
  if (c && h)
    return c == t && h == e;
  var v = -1, p = !0, $ = r & Br ? new q() : void 0;
  for (a.set(e, t), a.set(t, e); ++v < u; ) {
    var y = e[v], _ = t[v];
    if (n)
      var s = i ? n(_, y, v, t, e, a) : n(y, _, v, e, t, a);
    if (s !== void 0) {
      if (s)
        continue;
      p = !1;
      break;
    }
    if ($) {
      if (!Rr(t, function(l, b) {
        if (!Ur($, b) && (y === l || o(y, l, r, n, a)))
          return $.push(b);
      })) {
        p = !1;
        break;
      }
    } else if (!(y === _ || o(y, _, r, n, a))) {
      p = !1;
      break;
    }
  }
  return a.delete(e), a.delete(t), p;
}
var Gr = A.Uint8Array;
const W = Gr;
function Hr(e) {
  var t = -1, r = Array(e.size);
  return e.forEach(function(n, o) {
    r[++t] = [o, n];
  }), r;
}
function Kr(e) {
  var t = -1, r = Array(e.size);
  return e.forEach(function(n) {
    r[++t] = n;
  }), r;
}
var zr = 1, Yr = 2, qr = "[object Boolean]", Wr = "[object Date]", Zr = "[object Error]", Jr = "[object Map]", Xr = "[object Number]", Qr = "[object RegExp]", Vr = "[object Set]", kr = "[object String]", en = "[object Symbol]", tn = "[object ArrayBuffer]", rn = "[object DataView]", Se = L ? L.prototype : void 0, te = Se ? Se.valueOf : void 0;
function nn(e, t, r, n, o, a, i) {
  switch (r) {
    case rn:
      if (e.byteLength != t.byteLength || e.byteOffset != t.byteOffset)
        return !1;
      e = e.buffer, t = t.buffer;
    case tn:
      return !(e.byteLength != t.byteLength || !a(new W(e), new W(t)));
    case qr:
    case Wr:
    case Xr:
      return ce(+e, +t);
    case Zr:
      return e.name == t.name && e.message == t.message;
    case Qr:
    case kr:
      return e == t + "";
    case Jr:
      var u = Hr;
    case Vr:
      var f = n & zr;
      if (u || (u = Kr), e.size != t.size && !f)
        return !1;
      var c = i.get(e);
      if (c)
        return c == t;
      n |= Yr, i.set(e, t);
      var h = tt(u(e), u(t), n, o, a, i);
      return i.delete(e), h;
    case en:
      if (te)
        return te.call(e) == te.call(t);
  }
  return !1;
}
function rt(e, t) {
  for (var r = -1, n = t.length, o = e.length; ++r < n; )
    e[o + r] = t[r];
  return e;
}
function nt(e, t, r) {
  var n = t(e);
  return m(e) ? n : rt(n, r(e));
}
function an(e, t) {
  for (var r = -1, n = e == null ? 0 : e.length, o = 0, a = []; ++r < n; ) {
    var i = e[r];
    t(i, r, e) && (a[o++] = i);
  }
  return a;
}
function at() {
  return [];
}
var on = Object.prototype, sn = on.propertyIsEnumerable, Ee = Object.getOwnPropertySymbols, un = Ee ? function(e) {
  return e == null ? [] : (e = Object(e), an(Ee(e), function(t) {
    return sn.call(e, t);
  }));
} : at;
const de = un;
function fn(e, t) {
  for (var r = -1, n = Array(e); ++r < e; )
    n[r] = t(r);
  return n;
}
var cn = "[object Arguments]";
function je(e) {
  return j(e) && J(e) == cn;
}
var ot = Object.prototype, ln = ot.hasOwnProperty, gn = ot.propertyIsEnumerable, dn = je(function() {
  return arguments;
}()) ? je : function(e) {
  return j(e) && ln.call(e, "callee") && !gn.call(e, "callee");
};
const pn = dn;
function hn() {
  return !1;
}
var it = typeof exports == "object" && exports && !exports.nodeType && exports, xe = it && typeof module == "object" && module && !module.nodeType && module, vn = xe && xe.exports === it, Ce = vn ? A.Buffer : void 0, yn = Ce ? Ce.isBuffer : void 0, _n = yn || hn;
const Z = _n;
var bn = "[object Arguments]", $n = "[object Array]", Tn = "[object Boolean]", An = "[object Date]", wn = "[object Error]", On = "[object Function]", mn = "[object Map]", Pn = "[object Number]", Sn = "[object Object]", En = "[object RegExp]", jn = "[object Set]", xn = "[object String]", Cn = "[object WeakMap]", In = "[object ArrayBuffer]", Mn = "[object DataView]", Ln = "[object Float32Array]", Dn = "[object Float64Array]", Fn = "[object Int8Array]", Rn = "[object Int16Array]", Un = "[object Int32Array]", Nn = "[object Uint8Array]", Bn = "[object Uint8ClampedArray]", Gn = "[object Uint16Array]", Hn = "[object Uint32Array]", d = {};
d[Ln] = d[Dn] = d[Fn] = d[Rn] = d[Un] = d[Nn] = d[Bn] = d[Gn] = d[Hn] = !0;
d[bn] = d[$n] = d[In] = d[Tn] = d[Mn] = d[An] = d[wn] = d[On] = d[mn] = d[Pn] = d[Sn] = d[En] = d[jn] = d[xn] = d[Cn] = !1;
function Kn(e) {
  return j(e) && $t(e.length) && !!d[J(e)];
}
function pe(e) {
  return function(t) {
    return e(t);
  };
}
var st = typeof exports == "object" && exports && !exports.nodeType && exports, U = st && typeof module == "object" && module && !module.nodeType && module, zn = U && U.exports === st, re = zn && _t.process, Yn = function() {
  try {
    var e = U && U.require && U.require("util").types;
    return e || re && re.binding && re.binding("util");
  } catch {
  }
}();
const D = Yn;
var Ie = D && D.isTypedArray, qn = Ie ? pe(Ie) : Kn;
const ut = qn;
var Wn = Object.prototype, Zn = Wn.hasOwnProperty;
function ft(e, t) {
  var r = m(e), n = !r && pn(e), o = !r && !n && Z(e), a = !r && !n && !o && ut(e), i = r || n || o || a, u = i ? fn(e.length, String) : [], f = u.length;
  for (var c in e)
    (t || Zn.call(e, c)) && !(i && // Safari 9 has enumerable `arguments.length` in strict mode.
    (c == "length" || // Node.js 0.10 has enumerable non-index properties on buffers.
    o && (c == "offset" || c == "parent") || // PhantomJS 2 has enumerable non-index properties on typed arrays.
    a && (c == "buffer" || c == "byteLength" || c == "byteOffset") || // Skip index properties.
    Je(c, f))) && u.push(c);
  return u;
}
var Jn = Object.prototype;
function he(e) {
  var t = e && e.constructor, r = typeof t == "function" && t.prototype || Jn;
  return e === r;
}
function ct(e, t) {
  return function(r) {
    return e(t(r));
  };
}
var Xn = ct(Object.keys, Object);
const Qn = Xn;
var Vn = Object.prototype, kn = Vn.hasOwnProperty;
function ea(e) {
  if (!he(e))
    return Qn(e);
  var t = [];
  for (var r in Object(e))
    kn.call(e, r) && r != "constructor" && t.push(r);
  return t;
}
function ve(e) {
  return Xe(e) ? ft(e) : ea(e);
}
function ae(e) {
  return nt(e, ve, de);
}
var ta = 1, ra = Object.prototype, na = ra.hasOwnProperty;
function aa(e, t, r, n, o, a) {
  var i = r & ta, u = ae(e), f = u.length, c = ae(t), h = c.length;
  if (f != h && !i)
    return !1;
  for (var v = f; v--; ) {
    var p = u[v];
    if (!(i ? p in t : na.call(t, p)))
      return !1;
  }
  var $ = a.get(e), y = a.get(t);
  if ($ && y)
    return $ == t && y == e;
  var _ = !0;
  a.set(e, t), a.set(t, e);
  for (var s = i; ++v < f; ) {
    p = u[v];
    var l = e[p], b = t[p];
    if (n)
      var F = i ? n(b, l, p, t, e, a) : n(l, b, p, e, t, a);
    if (!(F === void 0 ? l === b || o(l, b, r, n, a) : F)) {
      _ = !1;
      break;
    }
    s || (s = p == "constructor");
  }
  if (_ && !s) {
    var P = e.constructor, S = t.constructor;
    P != S && "constructor" in e && "constructor" in t && !(typeof P == "function" && P instanceof P && typeof S == "function" && S instanceof S) && (_ = !1);
  }
  return a.delete(e), a.delete(t), _;
}
var oa = I(A, "DataView");
const oe = oa;
var ia = I(A, "Promise");
const ie = ia;
var sa = I(A, "Set");
const se = sa;
var ua = I(A, "WeakMap");
const ue = ua;
var Me = "[object Map]", fa = "[object Object]", Le = "[object Promise]", De = "[object Set]", Fe = "[object WeakMap]", Re = "[object DataView]", ca = C(oe), la = C(B), ga = C(ie), da = C(se), pa = C(ue), E = J;
(oe && E(new oe(new ArrayBuffer(1))) != Re || B && E(new B()) != Me || ie && E(ie.resolve()) != Le || se && E(new se()) != De || ue && E(new ue()) != Fe) && (E = function(e) {
  var t = J(e), r = t == fa ? e.constructor : void 0, n = r ? C(r) : "";
  if (n)
    switch (n) {
      case ca:
        return Re;
      case la:
        return Me;
      case ga:
        return Le;
      case da:
        return De;
      case pa:
        return Fe;
    }
  return t;
});
const G = E;
var ha = 1, Ue = "[object Arguments]", Ne = "[object Array]", z = "[object Object]", va = Object.prototype, Be = va.hasOwnProperty;
function ya(e, t, r, n, o, a) {
  var i = m(e), u = m(t), f = i ? Ne : G(e), c = u ? Ne : G(t);
  f = f == Ue ? z : f, c = c == Ue ? z : c;
  var h = f == z, v = c == z, p = f == c;
  if (p && Z(e)) {
    if (!Z(t))
      return !1;
    i = !0, h = !1;
  }
  if (p && !h)
    return a || (a = new T()), i || ut(e) ? tt(e, t, r, n, o, a) : nn(e, t, f, r, n, o, a);
  if (!(r & ha)) {
    var $ = h && Be.call(e, "__wrapped__"), y = v && Be.call(t, "__wrapped__");
    if ($ || y) {
      var _ = $ ? e.value() : e, s = y ? t.value() : t;
      return a || (a = new T()), o(_, s, r, n, a);
    }
  }
  return p ? (a || (a = new T()), aa(e, t, r, n, o, a)) : !1;
}
function lt(e, t, r, n, o) {
  return e === t ? !0 : e == null || t == null || !j(e) && !j(t) ? e !== e && t !== t : ya(e, t, r, n, lt, o);
}
function _a(e, t) {
  return lt(e, t);
}
function ba(e, t) {
  for (var r = -1, n = e == null ? 0 : e.length; ++r < n && t(e[r], r, e) !== !1; )
    ;
  return e;
}
function V(e, t, r, n) {
  var o = !r;
  r || (r = {});
  for (var a = -1, i = t.length; ++a < i; ) {
    var u = t[a], f = n ? n(r[u], e[u], u, r, e) : void 0;
    f === void 0 && (f = e[u]), o ? et(r, u, f) : ge(r, u, f);
  }
  return r;
}
function $a(e, t) {
  return e && V(t, ve(t), e);
}
function Ta(e) {
  var t = [];
  if (e != null)
    for (var r in Object(e))
      t.push(r);
  return t;
}
var Aa = Object.prototype, wa = Aa.hasOwnProperty;
function Oa(e) {
  if (!M(e))
    return Ta(e);
  var t = he(e), r = [];
  for (var n in e)
    n == "constructor" && (t || !wa.call(e, n)) || r.push(n);
  return r;
}
function ye(e) {
  return Xe(e) ? ft(e, !0) : Oa(e);
}
function ma(e, t) {
  return e && V(t, ye(t), e);
}
var gt = typeof exports == "object" && exports && !exports.nodeType && exports, Ge = gt && typeof module == "object" && module && !module.nodeType && module, Pa = Ge && Ge.exports === gt, He = Pa ? A.Buffer : void 0, Ke = He ? He.allocUnsafe : void 0;
function Sa(e, t) {
  if (t)
    return e.slice();
  var r = e.length, n = Ke ? Ke(r) : new e.constructor(r);
  return e.copy(n), n;
}
function Ea(e, t) {
  var r = -1, n = e.length;
  for (t || (t = Array(n)); ++r < n; )
    t[r] = e[r];
  return t;
}
function ja(e, t) {
  return V(e, de(e), t);
}
var xa = ct(Object.getPrototypeOf, Object);
const dt = xa;
var Ca = Object.getOwnPropertySymbols, Ia = Ca ? function(e) {
  for (var t = []; e; )
    rt(t, de(e)), e = dt(e);
  return t;
} : at;
const pt = Ia;
function Ma(e, t) {
  return V(e, pt(e), t);
}
function La(e) {
  return nt(e, ye, pt);
}
var Da = Object.prototype, Fa = Da.hasOwnProperty;
function Ra(e) {
  var t = e.length, r = new e.constructor(t);
  return t && typeof e[0] == "string" && Fa.call(e, "index") && (r.index = e.index, r.input = e.input), r;
}
function _e(e) {
  var t = new e.constructor(e.byteLength);
  return new W(t).set(new W(e)), t;
}
function Ua(e, t) {
  var r = t ? _e(e.buffer) : e.buffer;
  return new e.constructor(r, e.byteOffset, e.byteLength);
}
var Na = /\w*$/;
function Ba(e) {
  var t = new e.constructor(e.source, Na.exec(e));
  return t.lastIndex = e.lastIndex, t;
}
var ze = L ? L.prototype : void 0, Ye = ze ? ze.valueOf : void 0;
function Ga(e) {
  return Ye ? Object(Ye.call(e)) : {};
}
function Ha(e, t) {
  var r = t ? _e(e.buffer) : e.buffer;
  return new e.constructor(r, e.byteOffset, e.length);
}
var Ka = "[object Boolean]", za = "[object Date]", Ya = "[object Map]", qa = "[object Number]", Wa = "[object RegExp]", Za = "[object Set]", Ja = "[object String]", Xa = "[object Symbol]", Qa = "[object ArrayBuffer]", Va = "[object DataView]", ka = "[object Float32Array]", eo = "[object Float64Array]", to = "[object Int8Array]", ro = "[object Int16Array]", no = "[object Int32Array]", ao = "[object Uint8Array]", oo = "[object Uint8ClampedArray]", io = "[object Uint16Array]", so = "[object Uint32Array]";
function uo(e, t, r) {
  var n = e.constructor;
  switch (t) {
    case Qa:
      return _e(e);
    case Ka:
    case za:
      return new n(+e);
    case Va:
      return Ua(e, r);
    case ka:
    case eo:
    case to:
    case ro:
    case no:
    case ao:
    case oo:
    case io:
    case so:
      return Ha(e, r);
    case Ya:
      return new n();
    case qa:
    case Ja:
      return new n(e);
    case Wa:
      return Ba(e);
    case Za:
      return new n();
    case Xa:
      return Ga(e);
  }
}
var qe = Object.create, fo = function() {
  function e() {
  }
  return function(t) {
    if (!M(t))
      return {};
    if (qe)
      return qe(t);
    e.prototype = t;
    var r = new e();
    return e.prototype = void 0, r;
  };
}();
const co = fo;
function lo(e) {
  return typeof e.constructor == "function" && !he(e) ? co(dt(e)) : {};
}
var go = "[object Map]";
function po(e) {
  return j(e) && G(e) == go;
}
var We = D && D.isMap, ho = We ? pe(We) : po;
const vo = ho;
var yo = "[object Set]";
function _o(e) {
  return j(e) && G(e) == yo;
}
var Ze = D && D.isSet, bo = Ze ? pe(Ze) : _o;
const $o = bo;
var To = 1, Ao = 2, wo = 4, ht = "[object Arguments]", Oo = "[object Array]", mo = "[object Boolean]", Po = "[object Date]", So = "[object Error]", vt = "[object Function]", Eo = "[object GeneratorFunction]", jo = "[object Map]", xo = "[object Number]", yt = "[object Object]", Co = "[object RegExp]", Io = "[object Set]", Mo = "[object String]", Lo = "[object Symbol]", Do = "[object WeakMap]", Fo = "[object ArrayBuffer]", Ro = "[object DataView]", Uo = "[object Float32Array]", No = "[object Float64Array]", Bo = "[object Int8Array]", Go = "[object Int16Array]", Ho = "[object Int32Array]", Ko = "[object Uint8Array]", zo = "[object Uint8ClampedArray]", Yo = "[object Uint16Array]", qo = "[object Uint32Array]", g = {};
g[ht] = g[Oo] = g[Fo] = g[Ro] = g[mo] = g[Po] = g[Uo] = g[No] = g[Bo] = g[Go] = g[Ho] = g[jo] = g[xo] = g[yt] = g[Co] = g[Io] = g[Mo] = g[Lo] = g[Ko] = g[zo] = g[Yo] = g[qo] = !0;
g[So] = g[vt] = g[Do] = !1;
function Y(e, t, r, n, o, a) {
  var i, u = t & To, f = t & Ao, c = t & wo;
  if (r && (i = o ? r(e, n, o, a) : r(e)), i !== void 0)
    return i;
  if (!M(e))
    return e;
  var h = m(e);
  if (h) {
    if (i = Ra(e), !u)
      return Ea(e, i);
  } else {
    var v = G(e), p = v == vt || v == Eo;
    if (Z(e))
      return Sa(e, u);
    if (v == yt || v == ht || p && !o) {
      if (i = f || p ? {} : lo(e), !u)
        return f ? Ma(e, ma(i, e)) : ja(e, $a(i, e));
    } else {
      if (!g[v])
        return o ? e : {};
      i = uo(e, v, u);
    }
  }
  a || (a = new T());
  var $ = a.get(e);
  if ($)
    return $;
  a.set(e, i), $o(e) ? e.forEach(function(s) {
    i.add(Y(s, t, r, s, e, a));
  }) : vo(e) && e.forEach(function(s, l) {
    i.set(l, Y(s, t, r, l, e, a));
  });
  var y = c ? f ? La : ae : f ? ye : ve, _ = h ? void 0 : y(e);
  return ba(_ || e, function(s, l) {
    _ && (l = s, s = e[l]), ge(i, l, Y(s, t, r, l, e, a));
  }), i;
}
var Wo = 1, Zo = 4;
function ne(e) {
  return Y(e, Wo | Zo);
}
function Jo(e) {
  window.scrollTo({
    top: e - 200 + window.scrollY,
    behavior: "smooth"
  });
}
function ko(e) {
  const t = {}, r = Tt({ value: ne(e.initialValues), update: (/* @__PURE__ */ new Date()).getTime() }), n = k(!1), o = k({}), a = Ae(() => {
    var s;
    return Object.keys(o.value).length ? o.value : (s = e.validate) == null ? void 0 : s.call(e, r.value);
  }), i = Ae(() => !Object.keys(a.value).length), u = (s) => {
    _a(e.initialValues, r.value) || (s.returnValue = e.detectChangesMsg);
  };
  e.detectChangesMsg && (window.addEventListener("beforeunload", u), At(() => {
    window.removeEventListener("beforeunload", u);
  }));
  const f = () => {
    var P;
    if (!e.scrollToFirstError)
      return;
    let s = 0, l = "";
    (e.errorsResolver ? e.errorsResolver(a.value) : Object.keys(a.value)).forEach((S) => {
      var Te;
      const H = K(t, S);
      if (!H)
        return;
      const be = (Te = H == null ? void 0 : H.value) == null ? void 0 : Te.value;
      if (!be)
        return;
      const $e = be.getBoundingClientRect().top;
      ($e < s || !s) && (s = $e, l = S);
    }), (((P = e.beforeScroll) == null ? void 0 : P.call(e, l)) || !0) && s && Jo(s);
  }, c = (s) => {
    const l = {};
    for (const [b, F] of Object.entries(s))
      R(l, b, F);
    o.value = l, f();
  }, h = () => {
    n.value = !0, o.value = {}, r.update = (/* @__PURE__ */ new Date()).getTime(), i.value && (e.onSubmit(r.value), window.removeEventListener("beforeunload", u));
  };
  function v(s) {
    const l = k(null);
    return R(t, s, l), {
      ref: (b) => {
        l.value = b;
      },
      name: s,
      modelValue: K(r, `value.${s}`),
      error: n.value ? K(a.value, s) : null
    };
  }
  const p = (s) => K(a.value, s), $ = async (s, l) => {
    var b;
    R(r, `value.${l}`, s), R(o.value, l, void 0), (b = e.onUpdate) == null || b.call(e, l, s), e.submitOnChange && h();
  }, y = () => {
    R(r, "value", ne(e.initialValues)), o.value = {}, e.submitOnChange && h();
  }, _ = (s) => {
    r.value = ne(s);
  };
  return {
    value: r,
    submit: h,
    input: v,
    error: p,
    setResponseErrors: c,
    inputChangeHandler: $,
    isValid: i.value,
    reset: y,
    setValues: _,
    errors: a
  };
}
export {
  ko as u
};
