// @pfizer/hlx-ui/src/components/layout/TopServiceBar/TopNavBar.stories.ts
import type { Meta, StoryObj } from '@storybook/vue3';

import TopServiceBar from './TopServiceBar.vue';
import readme from './README.md?raw';

// More on how to set up stories at: https://storybook.js.org/docs/vue/writing-stories/introduction
const meta = {
  title: 'Layout/Top Service Bar',
  component: TopServiceBar,
  // This component will have an automatically generated docsPage entry: https://storybook.js.org/docs/vue/writing-docs/autodocs
  tags: ['autodocs'],
  parameters: {
    docs: {
      description: { component: readme },
    },
  },
  argTypes: {},
  args: {
    logo: 'https://placehold.co/120x60',
  }, // default value
  render: (args) => ({
    components: { TopServiceBar },
    setup() {
      return { args };
    },
    template: `
      <TopServiceBar v-bind="args" >
        <img src="https://placehold.co/24" alt=".">
        <img src="https://placehold.co/24" alt=".">
        <img src="https://placehold.co/24" alt=".">
      </TopServiceBar>
    `,
  }),
} satisfies Meta<typeof TopServiceBar>;

export default meta;
type Story = StoryObj<typeof meta>;
/*
 *👇 Render functions are a framework specific feature to allow you control on how the component renders.
 * See https://storybook.js.org/docs/vue/api/csf
 * to learn how to use render functions.
 */
export const Default: Story = {
  args: {},
};
